// opening_hours/site/js/i18n-resources.js
// Import i18next for use in helper functions
import i18next from '../../node_modules/i18next/dist/esm/i18next.bundled.js';

// localization {{{
export const resources = { // English is fallback language.
    // English (en) localization {{{
    en: {
        translation: {
            'lang': {
                'en': 'English',
                'fr': 'French',
                'de': 'German',
                'ru': 'Russian',
                'pt': 'Portuguese',
                'it': 'Italian',
                'uk': 'Ukrainian',
                'nl': 'Dutch',
                'hu': 'Hungarian',
                'es': 'Spanish',
                'vi': 'Vietnamese',
                'ja': 'Japanese',
                'choose': 'Choose Language',
            },

            'weekdays': {
                'day next week': 'next {{day}}'
            },
            'texts': {
                'filter': {
                    'none': 'Do not apply any filters',
                    'error': 'Error and warning messages',
                    'warnOnly': 'Warnings only',
                    'errorOnly': 'Error only',
                    'open': 'Only facilities which are open now',
                    'unknown': 'Only facilities which might be open now',
                    'closed': 'Only facilities which are closed now',
                    'openOrUnknown': 'Only facilities which are open or unknown now',
                },
                'title': 'opening_hours evaluation tool',
                'open always': 'Facility is always open',
                'unknown always': 'Facility is always maybe open',
                'closed always': 'Facility is always closed in the (near) future',
                'open now': 'Facility is now open',
                'unknown now': 'Facility might be open now',
                'closed now': 'Facility is now closed',
                'will close': 'but <a href="{{href}}">will</a> close {{timestring}}{{comment}}.',
                'will unknown': 'but <a href="{{href}}">will</a> maybe open {{timestring}}{{comment}}.',
                'will open': 'but <a href="{{href}}">will</a> open {{timestring}}{{comment}}.',
                'then close': 'and <a href="{{href}}">will be</a> closed {{timestring}}{{comment}}.',
                'then open': 'and <a href="{{href}}">will be</a> open {{timestring}}{{comment}}.',
                'depends on': ', but that depends on {{comment}}.',
                'reason': 'Reason',
                'opens again': 'Opens again',
                'closes again': 'Closes again',
                'jump to time': 'Jump to this time',
                'interval boundaries': 'Interval Boundaries (for developers)',
                'interval boundaries hint': 'Shows all points where an interval ends, even without state change.',
                'upcoming changes': 'Upcoming Changes',
                'week stable': 'Schedule is valid in any given week.',
                'not week stable': 'Attention! This schedule might change for other weeks.',
                'value for': 'value for',
                'value to compare': 'Compare with entered opening_hours value',
                'yohours incompatible': 'Value not compatible with YoHours editor',
                'MatchingRule': 'Applied rule',
                'prettified value': 'prettified opening_hours value (this value can be safely used in OSM, after all warnings have been <a href="{{copyFunc}}">solved</a>)',
                'prettified value for displaying': 'prettified opening_hours value for displaying (including newlines, do not use this as value for OSM)',
                'more information': 'For more information you can check out the <a href="{{href}}" target="_blank">OSM wiki</a>.',
                'this website': 'This website and the JavaScript library used for the evaluation of opening hours are developed on <a href="{{url}}" target="_blank">{{hoster}}</a>.',
                'if PH is between Mo and Fr': 'Only if the public holiday is a weekday (Mo-Fr)',
                'check out error correction, prettify': 'check out the error correction and the prettify function for the opening_hours value',
                'SH,PH or PH,SH': 'This makes a small difference compared to the previous value. The name of the school holidays will override the PH names in the comment.',
                'config POIs': 'configure POIs',
                'reload map': 'reload map',
                'heading map': 'Map with layer for the tag opening_hours based on OpenStreetMap',
                'map is showing': 'This map is showing objects with the tag <a rel="external" href="{{wikiUrl}}">opening_hours</a> as colored circles:',
                'error': 'The value could not be parsed',
                'warning': 'If there appeared warnings during evaluation, a blue {{sign}} will show up in the status icon.',
                'map filter': 'There are a few filters which can be applied to find and fix mistakes (QA) or to just display open or closed facilities:',
                'data source': 'The overlay data comes from {{OSMStartaTag}} and is queried using the {{APIaTag}}. The map is {{OSMaTag}}.',
                'mode 0': 'Only time ranges are accepted (tags opening_hours, lit)',
                'mode 1': 'Only points in time are accepted',
                'mode 2': 'Time ranges and points in time are accepted (tags service_times, collection_times)',
                'value to long for osm': 'The value is too long for OSM. The OpenStreetMap database is currently limited to values with up to {{maxLength}} characters. The prettified value has a length of {{pretLength}} and the value you entered is {{valLength}} characters long',
                'low zoom level': 'The POIs will start to appear at zoom level ${next} and above. You are currently on zoom level ${actual}.',
                'all n entries': 'All {{total}} entries:',
                'the first entries': 'The first {{number}} of {{total}} entries:',
                'load all with JOSM': 'load all in JOSM',
                'load osm objects': 'Load all OSM objects with this opening_hours value',
                'share config': 'Share this configuration',
                'with timestamp': 'With timestamp',
                'without timestamp': 'Without timestamp',
                'copy': 'Copy',
                'clear': 'Clear',
                'edit visually': 'Open in YoHours editor',
                'yohours description': 'A visual editor for simple opening hours syntax',
                'evaluation tool': 'evaluation tool',
                'rule separator ;': 'rule separator (the next rule will be a normal rule)',
                'rule separator  ||': 'rule separator (the next rule will be a fallback rule which applies for any time not handled by previous rules)',
                'rule separator ,': 'rule separator (the next rule will be a additional rule which extends the times of previous rules and does not override them like normal rules would do)',
                'JOSM remote conn error': 'Could not connect to JOSM. Please make sure that JOSM is running and is configured for remote control on the default tcp port 8111.',
                'Affected rules': 'Affected rules',
                'Original': 'Original',
                'Comparison': 'Comparison',
                'Rule': 'Rule',
                'Deviation at': 'Deviation at',
                'refer to yohours': 'This value can also be parsed by YoHours which is a simple editor for the opening_hours syntax. If you don’t need advanced features of the syntax, <a href="{{href}}">give it a try</a>.',
                'include timestamp?': 'include timestamp?',
            },
            'words': {
                'modifier': '{{name}} modifier',
                'selector': '{{name}} selector',
                'mode': 'evaluation mode',
                'green': 'green',
                'yellow': 'yellow',
                'red': 'red',
                'violet': 'violet',
                'from': 'from',
                'to': 'to',
                'and': 'and',
                'no': 'no',
                'undefined': 'undefined',
                'docu': 'documentation',
                'of course': 'of course',
                'open': 'open',
                'unknown': 'unknown',
                'closed': 'closed',
                'comment': 'comment',
                'today': 'today',
                'tomorrow': 'tomorrow',
                'yesterday': 'yesterday',
                'in duration': 'in',
                'region': 'region',
                'position': 'position',
                'lat': 'latitude',
                'lon': 'longitude',
                'country': 'country',
                'state': 'state',
                'status': 'status',
                'examples': 'Examples',
                'results': 'Results',
                'compare': 'Compare',
                'actions': 'Actions',
                'none': 'none',
                'date': 'date',
                'time': {
                    'minute': 'minute',
                    'minute_other': 'minutes',
                    'minute many': 'minutes',
                    'hour': 'hour',
                    'hour_other': 'hours',
                    'hour many': 'hours',
                    'day': 'day',
                    'day_other': 'days',
                    'day many': 'days',
                    'week': 'week',
                    'month': 'month',
                    'year': 'year',
                    'year_other': 'years',
                    'years many': 'years',
                    'hours minutes sep': 'and ',
                    'now': 'now',
                    'time': 'time'
                },
            },
        },
    }, // }}}

    es: {
        translation: {
            'lang': {
                'en': 'Inglés',
                'es': 'Español',
                'fr': 'Francés',
                'de': 'Alemán',
                'ru': 'Ruso',
                'pt': 'Portugués',
                'it': 'Italiano',
                'uk': 'Ucraniano',
                'nl': 'Neerlandés',
                'hu': 'Húngaro',
                'vi': 'Vietnamita',
                'ja': 'Idioma japonés',
                'choose': 'Elige idioma',
            },

            'weekdays': {
                'day next week': 'el próximo {{day}}'
            },
            'texts': {
                'filter': {
                    'none': 'No aplicar ningún filtro',
                    'error': 'Mensajes de error y advertencia',
                    'warnOnly': 'Solo advertencias',
                    'errorOnly': 'Solo errores',
                    'open': 'Solo instalaciones que están abiertas ahora',
                    'unknown': 'Solo instalaciones que pudieran estar abiertas ahora',
                    'closed': 'Solo instalaciones que están cerradas ahora',
                    'openOrUnknown': 'Solo instalaciones que están o pudieran estar abiertas ahora',
                },
                'title': 'herramienta de evaluación opening_hours',
                'open always': 'La instalación está siempre abierta',
                'unknown always': 'La instalación puede ser que esté siempre abierta',
                'closed always': 'La instalación está siempre cerrada en el futuro (cercano)',
                'open now': 'La instalación está abierta ahora',
                'unknown now': 'La instalación puede estar abierta ahora',
                'closed now': 'La instalación está ahora cerrada',
                'will close': 'pero <a href="{{href}}">cerrará</a> {{timestring}}{{comment}}.',
                'will unknown': 'pero <a href="{{href}}">puede ser</a> que abra {{timestring}}{{comment}}.',
                'will open': 'pero <a href="{{href}}">abrirá</a> {{timestring}}{{comment}}.',
                'then close': 'y <a href="{{href}}">estará</a> cerrado {{timestring}}{{comment}}.',
                'then open': 'y <a href="{{href}}">estará</a> abierto {{timestring}}{{comment}}.',
                'depends on': ', pero eso depende de {{comment}}.',
                'reason': 'Motivo',
                'opens again': 'Abre de nuevo',
                'closes again': 'Cierra de nuevo',
                'jump to time': 'Ir a esta hora',
                'interval boundaries': 'Límites de intervalo (para desarrolladores)',
                'interval boundaries hint': 'Muestra todos los puntos donde termina un intervalo, incluso sin cambio de estado.',
                'upcoming changes': 'Próximos Cambios',
                'week stable': 'El horario es válido en cualquier semana dada.',
                'not week stable': '¡Atención! Este horario puede ser distinto para otras semanas.',
                'value for': 'valor para',
                'value to compare': 'Comparar con el valor de opening_hours introducido',
                'yohours incompatible': 'Valor no compatible con el editor YoHours',
                'MatchingRule': 'Regla aplicada',
                'prettified value': 'valor embellecido de opening_hours (este valor puede ser usado con seguridad en OSM, después de haber <a href="{{copyFunc}}">resuelto</a> todas las advertencias)',
                'prettified value for displaying': 'valor embellecido de opening_hours para mostrar (incluyendo saltos de línea, no use este valor en OSM)',
                'more information': 'Para más información, puede consultar la <a href="{{href}}" target="_blank">wiki OSM</a>.',
                'this website': 'Esta página web y la librería JavaScript usadas para la evaluación de las horas de apertura han sido desarrolladas en <a href="{{url}}" target="_blank">{{hoster}}</a>.',
                'if PH is between Mo and Fr': 'Sólo si el día festivo está entre el lunes y el viernes',
                'check out error correction, prettify': 'compruebe la corrección de error y la función embellecedora para el valor de opening_hours',
                'SH,PH or PH,SH': 'Esto hace una pequeña diferencia comparado con el valor anterior. El nombre de los festivos de escuela sobreescribirá los nombres de festivos públicos en el comentario.',
                'config POIs': 'configurar PDIs',
                'reload map': 'actualizar mapa',
                'heading map': 'Mapa con capa para la etiqueta opening_hours basado en OpenStreetMap',
                'map is showing': 'Este mapa está mostrando objetos con la etiqueta <a rel="external" href="{{wikiUrl}}">opening_hours</a> como círculos coloreados:',
                'error': 'El valor no ha podido ser analizado',
                'warning': 'Si han aparecido advertencias durante la evaluación, una {{sign}} azul aparecerá en el icono de estado.',
                'map filter': 'Hay algunos filtros que pueden ser aplicados para encontrar y arreglar fallos (QA) o para simplemente mostrar instalaciones abiertas o cerradas:',
                'data source': 'Los datos superpuestos provienen de {{OSMStartaTag}} y han sido consultados usando la {{APIaTag}}. El mapa es {{OSMaTag}}.',
                'mode 0': 'Solo rangos de tiempo son aceptados (etiquetas opening_hours, lit)',
                'mode 1': 'Solo puntos en el tiempo son aceptados',
                'mode 2': 'Rangos de tiempo y puntos en el tiempo son aceptados (etiquetas service_times, collection_times)',
                'value to long for osm': 'El valor es demasiado largo para OSM. La base de datos de OpenStreetMap está actualmente limitada a valores de hasta {{maxLength}} caracteres. El valor embellecido tiene una longitud de {{pretLength}} y el valor que ha introducido tiene {{valLength}} caracteres de longitud',
                'low zoom level': 'Los PDIs empezarán a aparecer en el nivel de zoom ${next} y superiores. Está actualmente en el nivel ${actual}.',
                'all n entries': 'Todas las {{total}} entradas:',
                'the first entries': 'Las primeras {{number}} of {{total}} entradas:',
                'load all with JOSM': 'cargar todo en JOSM',
                'load osm objects': 'Cargar todos los objetos OSM con este valor de opening_hours',
                'share config': 'Compartir esta configuración',
                'with timestamp': 'Con marca de tiempo',
                'without timestamp': 'Sin marca de tiempo',
                'copy': 'Copiar',
                'clear': 'Borrar',
                'edit visually': 'Abrir en el editor YoHours',
                'yohours description': 'Un editor visual para sintaxis de opening_hours sencilla',
                'evaluation tool': 'herramienta de evaluación',
                'rule separator ;': 'searador de reglas (la siguiente regla será una regla normal)',
                'rule separator  ||': 'separador de reglas (la siguiente regla será una regla de reserva que será aplicada para cualquier tiempo que no haya sido manipulado por las reglas anteriores)',
                'rule separator ,': 'separador de reglas (la siguiente regla será una regla adicional que extiende los tiempos de las reglas anteriores y no las sobreescribe como harían las reglas normales)',
                'JOSM remote conn error': 'No pudo conectar a JOSM. Por favor, asegúrese de que JOSM está corriendo y está configurado para control remoto en el puerto tcp por defecto 8111.',
                'refer to yohours': 'Este valor puede ser analizado también por YoHours, que es un editor simpre para la sintaxis de opening_hours. Si no necesita funcionalidades avanzadas de la sintaxis, <a href="{{href}}">puede darle una oportunidad</a>.',
                'include timestamp?': '¿incluir marca de tiempo?',
            },
            'words': {
                'modifier': 'modificador {{name}}',
                'selector': 'selector {{name}}',
                'mode': 'modo evaluación',
                'green': 'verde',
                'yellow': 'amarillo',
                'red': 'rojo',
                'violet': 'violeta',
                'from': 'de',
                'to': 'a',
                'and': 'y',
                'no': 'no',
                'undefined': 'indefinido',
                'docu': 'documentación',
                'of course': 'por supuesto',
                'open': 'abierto',
                'unknown': 'desconocido',
                'closed': 'cerrado',
                'comment': 'comentario',
                'today': 'hoy',
                'tomorrow': 'mañana',
                'yesterday': 'ayer',
                'in duration': 'en',
                'region': 'región',
                'position': 'posición',
                'lat': 'latitud',
                'lon': 'longitud',
                'country': 'país',
                'state': 'estado',
                'status': 'estado',
                'examples': 'Ejemplos',
                'results': 'Resultados',
                'compare': 'Comparar',
                'actions': 'Acciones',
                'none': 'ninguno',
                'date': 'fecha',
                'time': {
                    'minute': 'minuto',
                    'minute_other': 'minutos',
                    'minute many': 'minutos',
                    'hour': 'hora',
                    'hour_other': 'horas',
                    'hour many': 'horas',
                    'day': 'día',
                    'day_other': 'días',
                    'day many': 'días',
                    'week': 'semana',
                    'month': 'mes',
                    'year': 'año',
                    'year_other': 'años',
                    'years many': 'años',
                    'hours minutes sep': 'y ',
                    'now': 'ahora',
                    'time': 'hora',
                },
            },
        },
    }, // }}}

    // Dutch (nl) localization {{{
    nl: {
        translation: {
            'lang': {
                'en': 'Engels',
                'fr': 'Frans',
                'de': 'Duits',
                'ru': 'Russisch',
                'pt': 'Portugees',
                'it': 'Italiaans',
                'uk': 'Oekraïens',
                'nl': 'Nederlands',
                'hu': 'Hongaars',
                'es': 'Spaans',
                'vi': 'Vietnamees',
                'ja': 'Japans',
                'choose': 'Kies taal',
            },

            'weekdays': {
                'day next week': 'volgende {{day}}'
            },
            'texts': {
                'filter': {
                    'none': 'Geen filters toepassen',
                    'error': 'Fouten en waarschuwingen',
                    'warnOnly': 'Enkel waarschuwingen',
                    'errorOnly': 'Enkel foutboodschappen',
                    'open': 'Enkel zaken die nu open zijn',
                    'unknown': 'Enkel zaken die nu mogelijks open zijn',
                    'closed': 'Enkel zaken die nu gesloten zijn',
                    'openOrUnknown': 'Enkel zaken die nu open zijn of die mogelijks open zijn',
                },
                'title': 'opening_hours Evaluatie Gereedschap',
                'open always': 'Zaak is altijd open',
                'unknown always': 'Zaak is waarschijnlijk altijd open',
                'closed always': 'Zaak is altijd gesloten in de (nabije) toekomst',
                'open now': 'Zaak is nu open',
                'unknown now': 'Zaak zou nu open kunnen zijn',
                'closed now': 'Zaak is nu gesloten',
                'will close': 'maar <a href="{{href}}">zal</a> sluiten op {{timestring}}{{comment}}.',
                'will unknown': 'maar <a href="{{href}}">zal</a> misschien open zjn op {{timestring}}{{comment}}.',
                'will open': 'maar <a href="{{href}}">zal</a> openen op {{timestring}}{{comment}}.',
                'then close': 'en <a href="{{href}}">zal</a> gesloten zijn {{timestring}}{{comment}}.',
                'then open': 'en <a href="{{href}}">zal</a> open zijn {{timestring}}{{comment}}.',
                'depends on': ', maar hangt af van {{comment}}.',
                'reason': 'Reden',
                'opens again': 'Opent weer',
                'closes again': 'Sluit weer',
                'jump to time': 'Naar deze tijd springen',
                'interval boundaries': 'Intervalgrenzen (voor ontwikkelaars)',
                'interval boundaries hint': 'Toont alle punten waar een interval eindigt, ook zonder statuswijziging.',
                'upcoming changes': 'Aankomende Wijzigingen',
                'week stable': 'Openingsuren zijn altijd geldig.',
                'not week stable': 'Opegelet! Openingsuren kunnen wijzigen afhankelijk van de gekozen week.',
                'value for': 'waarde voor',
                'value to compare': 'Vergelijk met ingevoerde opening_hours waarde',
                'yohours incompatible': 'Waarde niet compatibel met YoHours editor',
                'MatchingRule': 'Toegepaste regel',
                'prettified value': 'Mooi gemaakte opening_hours waarde (deze waarde kan zonder problemen in OSM gebruikt worden, nadat alle waarschuwingen zijn  <a href="{{copyFunc}}">weggewerkt</a>)',
                'prettified value for displaying': 'Mooi gemaakte opening_hours waarde om op het scherm te tonen (bevat verschillende lijnen, gebruik dit niet als waarde voor OSM)',
                'more information': 'Voor meer informatie, raadpleef de <a href="{{href}}" target="_blank">OSM wiki</a>.',
                'this website': 'Deze website en de JavaScript library die gebruikt wordt voor de evaluatie zijn ontwikkeld op <a href="{{url}}" target="_blank">{{hoster}}</a>.',
                'if PH is between Mo and Fr': 'Enkel als de feestdag op een weekdag valt (Ma-Vr)',
                'check out error correction, prettify': 'controleer de fout correctie en de mooi maak functie voor de opening_hours waarde',
                'SH,PH or PH,SH': 'Dit maakt een klein verschil met de vorige waarde. De naam van de vrije schooldagen zal de namen van de feestdagen in de commentaar overschrijven.',
                'config POIs': 'configureer POIs',
                'reload map': 'herlaad kaart',
                'heading map': 'Kaart met laag voor de opening_hours tag op on OpenStreetMap',
                'map is showing': 'Deze kaart toont de object met tag <a rel="external" href="{{wikiUrl}}">opening_hours</a> als gekleurde cirkels:',
                'error': 'De waarde kon niet ontleed worden',
                'warning': 'Indien er waarschuwingen getoond werden tijden de evaluatie, zal een blauw {{sign}} getoond worden in het status icoon.',
                'map filter': 'Er zijn een aantal filters die kunnen toegepast worden om fouten te vinden en te verbeteren (QA) of gewoon om gesloten of open zaken te tonen:',
                'data source': 'De overlay gegevens komen van {{OSMStartaTag}} en is bevraagd met {{APIaTag}}. De kaart is {{OSMaTag}}.',
                'mode 0': 'Alleen tijdsperioden zijn toegestaan (zoals bij de tags opening_hours, lit)',
                'mode 1': 'Individuele tijdspunten zijn toegestaan',
                'mode 2': 'Zowel tijdsperioden als individuele tijdspunten zijn toegestaan (zoals bij de tags service_times, collection_times)',
                'value to long for osm': 'De waarde is te lang voor OSM. De OpenStreetMap databank is momenteel beperkt tot waardes met {{maxLength}} karakter. De mooi gemaakte waarde heeft een lengte van {{pretLength}} en de ingevoerde waarde is {{valLength}} karakters lang',
                'low zoom level': 'De POIs verschijnen vanaf zoomniveau ${next}. U bent momenteel op zoomniveau ${actual}.',
                'all n entries': 'Alle resultaten ({{total}}):',
                'the first entries': 'De eerste {{number}} van {{total}} resultaten:',
                'load all with JOSM': 'laad alles in JOSM',
                'load osm objects': 'Laad alle OSM-objecten met deze opening_hours-waarde',
                'share config': 'Deel deze configuratie',
                'with timestamp': 'Met tijdstempel',
                'without timestamp': 'Zonder tijdstempel',
                'copy': 'Kopiëren',
                'clear': 'Wissen',
                'edit visually': 'Open in YoHours-editor',
                'yohours description': 'Een visuele editor voor eenvoudige opening_hours syntax',
                'evaluation tool': 'evaluatie gereedschap',
                'rule separator ;': 'scheidingsteken voor regel (de volgende regel zal een normale regel zijn)',
                'rule separator  ||': 'scheidingsteken voor regel (de volgende regel will be a fallback rule which applies for any time not handled by previous rules)',
                'rule separator ,': 'scheidingsteken voor regel (de volgende regel is aanvullende regel die de tijden van de vorige regels uitbreid. De volgende regel overschrijft de vorige regels dus niet (zoals een normale regel wel doet)',
                'JOSM remote conn error': 'Kan niet connecteren met JOSM. Controleer aub dat  JOSM is opgestart en geconfigureerd is voor afstandbediening via de standaard tcp port 8111.',
                'Affected rules': 'Betrokken regels',
                'Original': 'Origineel',
                'Comparison': 'Vergelijking',
                'Rule': 'Regel',
                'Deviation at': 'Afwijking op',
                'refer to yohours': 'Deze waarde kan ook geïnterpreteerd worden door YoHours. Dit is een eenvoudige editor voor de opening_hours syntax.  Indien je de geavanceerde mogelijkheden van de syntax niet nodig hebt, <a href="{{href}}">probeer YoHours dan eens</a>.',
                'include timestamp?': 'tijdstempel opnemen?',
            },
            'words': {
                'modifier': '{{name}} wijziger (modifier)',
                'selector': '{{name}} keuzeschakelaar (selector)',
                'mode': 'evaluatie modus',
                'green': 'groen',
                'yellow': 'geel',
                'red': 'rood',
                'violet': 'violet',
                'from': '',
                'to': 'tot',
                'and': 'en',
                'no': 'nee',
                'docu': 'documentatie',
                'of course': 'natuurlijk',
                'open': 'open',
                'unknown': 'onbekend',
                'closed': 'gesloten',
                'comment': 'commentaar',
                'today': 'vandaag',
                'tomorrow': 'morgen',
                'yesterday': 'hier',
                'in duration': 'in',
                'region': 'regio',
                'position': 'positie',
                'lat': 'latitude',
                'lon': 'longitude',
                'country': 'land',
                'state': 'staat',
                'status': 'status',
                'examples': 'Voorbeelden',
                'results': 'Resultaten',
                'compare': 'Vergelijken',
                'actions': 'Acties',
                'none': 'geen',
                'date': 'datum',
                'time': {
                    'minute': 'minuut',
                    'minute_other': 'minuten',
                    'minute many': 'minuten',
                    'hour': 'uur',
                    'hour_other': 'uren',
                    'hour many': 'uren',
                    'day': 'dag',
                    'day_other': 'dagen',
                    'day many': 'dagen',
                    'week': 'week',
                    'month': 'maand',
                    'year': 'jaar',
                    'year_other': 'jaren',
                    'years many': 'jaren',
                    'hours minutes sep': 'en ',
                    'now': 'nu',
                    'time': 'tijd',
                },
                'undefined': 'ongedefinieerd',
            },
        },
    }, // }}}

    // French (fr) localization {{{
    fr: {
        translation: {
            'lang': {
                'en': 'Anglais',
                'fr': 'Français',
                'de': 'Allemand',
                'ru': 'Russe',
                'pt': 'Portugais',
                'it': 'Italien',
                'uk': 'Ukrainien',
                'nl': 'Néerlandais',
                'hu': 'Hongrois',
                'es': 'Espagnol',
                'vi': 'Vietnamien',
                'ja': 'Japonais',
                'choose': 'Choisissez la langue',
            },

            'weekdays': {
                'day next week': '{{day}} prochain'
            },
            'texts': {
                'filter': {
                    'none': 'Do not apply any filters',
                    'error': 'Messages d\'erreur et d\'avertissement',
                    'warnOnly': 'Seulement les avertissements',
                    'errorOnly': 'Seulement les erreurs',
                    'open': 'Seulement les installations ouvertes en ce moment',
                    'unknown': 'Seulement les installations indéterminées en ce moment',
                    'closed': 'Seulement les installations fermées en ce moment',
                    'openOrUnknown': 'Seulement les installations ouvertes ou indéterminées en ce moment',
                },
                'open always': 'L\'installation est ouverte en permanence',
                'unknown always': 'L\'installation est peut-être ouverte en permanence',
                'closed always': 'L\'installation est fermée en permanence (à court terme)',
                'open now': 'L\'installation est ouverte en ce moment',
                'unknown now': 'L\'installation est indéterminée en ce moment',
                'closed now': 'L\'installation est fermée en ce moment',
                'will close': 'mais <a href="{{href}}">va</a> fermer {{timestring}}{{comment}}.',
                'will unknown': 'mais <a href="{{href}}">va</a> peut-être ouvrir {{timestring}}{{comment}}.',
                'will open': 'mais <a href="{{href}}">va</a> ouvrir {{timestring}}{{comment}}.',
                'then close': 'et <a href="{{href}}">sera</a> fermé {{timestring}}{{comment}}.',
                'then open': 'et <a href="{{href}}">sera</a> ouvert {{timestring}}{{comment}}.',
                'depends on': ', mais cela dépend de {{comment}}.',
                'reason': 'Raison',
                'opens again': 'Ouvre à nouveau',
                'closes again': 'Ferme à nouveau',
                'jump to time': 'Aller à cette heure',
                'interval boundaries': 'Limites d\'intervalle (pour développeurs)',
                'interval boundaries hint': 'Affiche tous les points où un intervalle se termine, même sans changement d\'état.',
                'upcoming changes': 'Prochains Changements',
                'week stable': 'L\'horaire est valable dans n\'importe quelle semaine donnée.',
                'not week stable': 'Attention ! Cet horaire peut changer pour les autres semaines.',
                'value for': 'valeur pour',
                'value to compare': 'Comparer avec la valeur opening_hours saisie',
                'yohours incompatible': 'Valeur non compatible avec l\'éditeur YoHours',
                'MatchingRule': 'Sous-chaîne utilisée par la règle appliquée',
                'prettified value': 'valeur opening_hours embellie (cette valeur peut être utilisée en toute sécurité dans OSM, après que tous les avertissements aient été <a href="{{copyFunc}}">corrigés</a>)',
                'prettified value for displaying': 'valeur opening_hours embellie pour l\'affichage (inclut des sauts de ligne, ne pas l\'utiliser comme valeur pour OSM)',
                'more information': 'Pour plus d\'informations vous pouvez consulter le <a href="{{href}}" target="_blank">wiki OSM</a>.',
                'this website': 'Ce site web et la bibliothèque JavaScript utilisée pour l\'évaluation des horaires d\'ouverture sont développés sur <a href="{{url}}" target="_blank">{{hoster}}</a>.',
                'if PH is between Mo and Fr': 'Seulement si le jour férié est un jour de semaine (lundi-vendredi)',
                'check out error correction, prettify': 'consultez la correction d\'erreur et la fonction prettify pour la valeur de opening_hours',
                'SH,PH or PH,SH': 'Ceci donne un petit écart par rapport à la valeur précédente. Le nom des vacances scolaires remplace les noms des jours fériés dans le commentaire.',
                'config POIs': 'configurer les POIs',
                'reload map': 'rafraîchir la carte',
                'heading map': 'Carte avec calque pour le tag opening_hours basée sur OpenStreetMap',
                'map is showing': 'cette carte affiche les nœuds portant le tag <a rel="external" href="{{wikiUrl}}">opening_hours</a> comme des cercles colorés :',
                'error': 'La valeur n\'a pas pu être analysée',
                'warning': 'S\'il y a eu des avertissements lors de l\'évaluation, un {{sign}} bleu s\'affichera dans l\'icône d\'état.',
                'map filter': 'Il ya quelques filtres qui peuvent être appliqués pour trouver et corriger les erreurs (AQ) ou pour afficher seulement les installations ouvertes ou fermées:',
                'data source': 'Les données en superposition viennent de {{APIaTag}}. La carte est {{OSMaTag}}.',
                'mode 0': 'Seuls les intervalles de temps sont acceptés (tags opening_hours, lit)',
                'mode 1': 'Seuls les points dans le temps sont acceptés',
                'mode 2': 'Les plages horaires et les points dans le temps sont acceptés (tags service_times, collection_times)',
                'value to long for osm': 'La valeur est trop longue pour OSM. La base de données OpenStreetMap est actuellement limitée à des valeurs d\'un maximum de {{maxLength}} caractères. La valeur embellie a une longueur de {{pretLength}} et la valeur que vous avez entrée comporte {{valLength}} caractères',
                'low zoom level': 'Les POIs vont commencer à apparaître au niveau de zoom ${next} et au-delà. Vous êtes actuellement au niveau de zoom ${actual}.',
                'all n entries': 'Ensemble des {{total}} entrées :',
                'the first entries': 'Les {{number}} premières entrées sur {{total}} :',
                'Affected rules': 'Règles affectées',
                'Original': 'Original',
                'Comparison': 'Comparaison',
                'Rule': 'Règle',
                'Deviation at': 'Déviation à',
                'refer to yohours': 'Cette valeur peut également s\'afficher dans YoHours, un éditeur simple d\'horaires d\'ouverture. Si vous n\'utilisez pas les fonctionnalités avancées de la syntaxe, <a href="{{href}}">vous pouvez l\'essayer ici</a>.',
                'load all with JOSM': 'tout charger dans JOSM',
                'load osm objects': 'Charger tous les objets OSM avec cette valeur opening_hours',
                'share config': 'Partager cette configuration',
                'with timestamp': 'Avec horodatage',
                'without timestamp': 'Sans horodatage',
                'copy': 'Copier',
                'clear': 'Effacer',
                'edit visually': 'Ouvrir dans l\'éditeur YoHours',
                'yohours description': 'Un éditeur visuel pour une syntaxe d\'horaires simple',
                'evaluation tool': 'outil d\'évaluation',
                'title': 'outil d\'évaluation opening_hours',
                'rule separator ;': 'séparateur de règle (la règle suivante sera une règle normale)',
                'rule separator  ||': 'séparateur de règle (la règle suivante sera une règle de repli qui s\'applique à tout moment non géré par les règles précédentes)',
                'rule separator ,': 'séparateur de règle (la règle suivante sera une règle additionnelle qui étend les horaires des règles précédentes sans les remplacer)',
                'JOSM remote conn error': 'Impossible de se connecter à JOSM. Veuillez vous assurer que JOSM est en cours d\'exécution et configuré pour le contrôle à distance sur le port tcp 8111 par défaut.',
                'include timestamp?': 'inclure l\'horodatage ?',
            },
            'words': {
                'modifier': 'modificateur {{name}}',
                'selector': 'sélecteur {{name}}',
                'mode': 'mode d\'évaluation',
                'green': 'vert',
                'yellow': 'jaune',
                'red': 'rouge',
                'violet': 'violet',
                'from': '',
                'to': 'à',
                'and': 'et',
                'no': 'pas de',
                'docu': 'documentation',
                'of course': 'bien sûr',
                'open': 'ouvert',
                'unknown': 'indéterminé',
                'closed': 'fermé',
                'comment': 'commentaire',
                'today': 'aujourd\'hui',
                'tomorrow': 'demain',
                'yesterday': 'hier',
                'in duration': 'dans',
                'region': 'région',
                'position': 'emplacement',
                'lat': 'latitude',
                'lon': 'longitude',
                'country': 'pays',
                'state': 'état',
                'status': 'statut',
                'examples': 'Exemples',
                'results': 'Résultats',
                'compare': 'Comparer',
                'actions': 'Actions',
                'none': 'aucun',
                'date': 'date',
                'time': {
                    'minute': 'minute',
                    'minute_other': 'minutes',
                    'minute many': 'minutes',
                    'hour': 'heure',
                    'hour_other': 'heures',
                    'hour many': 'heures',
                    'day': 'jour',
                    'day_other': 'jours',
                    'day many': 'jours',
                    'week': 'semaine',
                    'month': 'mois',
                    'year': 'an',
                    'year_other': 'ans',
                    'years many': 'ans',
                    'hours minutes sep': 'et ',
                    'now': 'en ce moment',
                    'time': 'heure',
                },
                'undefined': 'indéfini'
            },
        },
    }, // }}}

    // German (de) localization {{{
    de: {
        translation: {
            'lang': {
                'en': 'Englisch',
                'fr': 'Französisch',
                'de': 'Deutsch',
                'ru': 'Russisch',
                'pt': 'Portugiesisch',
                'it': 'Italienisch',
                'uk': 'Ukrainisch',
                'nl': 'Niederländisch',
                'hu': 'Ungarisch',
                'es': 'Spanisch',
                'vi': 'Vietnamesisch',
                'ja': 'Japanisch',
                'choose': 'Wähle eine Sprache',
            },

            'weekdays': {
                'day next week': 'nächsten {{day}}'
            },
            'texts': {
                'filter': {
                    'none': 'Keinen Filter anwenden',
                    'error': 'Fehler und Warnungen',
                    'warnOnly': 'Nur Warnungen',
                    'errorOnly': 'Nur Fehler',
                    'open': 'Nur Einrichtungen welche jetzt geöffnet sind',
                    'unknown': 'Nur Einrichtungen die möglicherweise jetzt geöffnet sind',
                    'closed': 'Nur Einrichtungen die jetzt geschlossen sind',
                    'openOrUnknown': 'Nur Einrichtungen die jetzt geöffnet oder unbekannt sind',
                },
                'title': 'opening_hours Auswertewerkzeug',
                'open always': 'Die Einrichtung hat immer geöffnet.',
                'unknown always': 'Die Öffnungszeit der Einrichtung ist immer unbekannt',
                'closed always': 'Die Einrichtung wird in (naher) Zukunft nicht wieder öffnen.',
                'open now': 'Die Einrichtung hat jetzt geöffnet',
                'unknown now': 'Die Einrichtung ist möglicherweise jetzt geöffnet',
                'closed now': 'Die Einrichtung ist jetzt geschlossen',
                'will close': 'aber <a href="{{href}}">wird</a> {{timestring}} schließen{{comment}}.',
                'will unknown': 'aber <a href="{{href}}">wird</a> eventuell {{timestring}} öffnen{{comment}}.',
                'will open': 'aber <a href="{{href}}">wird</a> {{timestring}} öffnen{{comment}}.',
                'then close': 'und <a href="{{href}}">ist dann</a> {{timestring}} geschlossen{{comment}}.',
                'then open': 'und <a href="{{href}}">ist dann</a> {{timestring}} geöffnet{{comment}}.',
                'depends on': ', abhängig von {{comment}}.',
                'reason': 'Grund',
                'opens again': 'Öffnet wieder',
                'closes again': 'Schließt wieder',
                'jump to time': 'Zu dieser Zeit springen',
                'interval boundaries': 'Intervallgrenzen (für Entwickler)',
                'interval boundaries hint': 'Zeigt alle Zeitpunkte, an denen ein Intervall endet, auch ohne Zustandswechsel.',
                'upcoming changes': 'Kommende Änderungen',
                'week stable': 'Dieser Wochenplan gilt für jede Woche.',
                'not week stable': 'Achtung! Dieser Wochenplan kann sich in anderen Wochen ändern.',
                'value for': 'Wert für',
                'value to compare': 'Mit eingegebenem opening_hours-Wert vergleichen',
                'yohours incompatible': 'Wert nicht mit YoHours-Editor kompatibel',
                'MatchingRule': 'Zur Anwendung gekommene Regel',
                'prettified value': 'Schön formatierter opening_hours Wert (dieser Wert sollte in OSM verwendet werden, nachdem alle Warnungen <a href="{{copyFunc}}">beseitigt wurden</a>)',
                'prettified value for displaying': 'Schön formatierter opening_hours Wert für die Anzeige (mit Zeilenumbrüchen, nicht als Tag für OSM gedacht …)',
                'more information': 'Für weitere Informationen kannst du dir das <a href="{{href}}" target="_blank">OSM-Wiki</a> anschauen.',
                'this website': 'Die Entwicklung dieser Webseite und der JavaScript-Bibliothek zur Auswertung der Öffnungszeiten findet auf <a href="{{url}}" target="_blank">{{hoster}}</a> statt.',
                'if PH is between Mo and Fr': 'Nur wenn der Feiertag auf einen Wochentag (Mo-Fr) fällt',
                'check out error correction, prettify': 'Probiere die Fehlererkennung und schau dir den korrigierten opening_hours Wert an',
                'SH,PH or PH,SH': 'Dieser Wert unterscheidet sich insofern zum vorherigen als dass der Name für Schulferien den Namen von Feiertagen im Kommentar überschreibt, wenn sich diese überlagern',
                'config POIs': 'POIs konfigurieren',
                'reload map': 'Karte aktualisieren',
                'heading map': 'Karte zur Auswertung des Tags opening_hours basierend auf OpenStreetMap',
                'map is showing': 'Diese Karte zeigt Objekte mit dem Tag <a rel="external" href="{{wikiUrl}}">opening_hours</a> als farbige Kreise:',
                'error': 'Die Öffnungszeit konnte nicht ausgewertet werden',
                'warning': 'Sollte eine Öffnungszeit beim verarbeiten eine Warnung erzeugen, wird im Status Icon ein blaues {{sign}} angezeigt.',
                'map filter': 'Es gibt mehrere Filter um nur die Öffnungszeiten von Interesse anzuzeigen. Die Hauptgründe für die Verwendung eines Filters ist die Qualitätssicherung oder das finden von offenen Geschäften. Es folgt eine Liste der möglichen Filter:',
                'data source': 'Die auszuwertenden Daten stammen aus {{OSMStartaTag}} und werden über die {{APIaTag}} abgefragt. Die Karte ist {{OSMaTag}}.',
                'mode 0': 'Es werden nur Zeiträume akzeptiert (Tags opening_hours, lit)',
                'mode 1': 'Es werden nur Zeitpunkte akzeptiert',
                'mode 2': 'Zeiträume und Zeitpunkte werden akzeptiert (Tags service_times, collection_times)',
                'value to long for osm': 'Der opening_hours Wert ist zu lang für OSM. Die OpenStreetMap-Datenbank unterstützt zur Zeit nur Werte mit bis zu {{maxLength}} Zeichen. Der schön formatierte Wert ist {{pretLength}} Zeichen lang und der eingegebene Wert {{valLength}}.',
                'low zoom level': 'Die Marker werden erst auf Zoom-Stufe ${next} geladen. Aktuelle Zoom-Stufe ist ${actual}.',
                'all n entries': 'Alle {{total}} Einträge:',
                'the first entries': 'Die ersten {{number}} von {{total}} Einträgen:',
                'load all with JOSM': 'Alle in JOSM laden',
                'load osm objects': 'Alle OSM-Objekte mit diesem opening_hours-Wert laden',
                'share config': 'Diese Konfiguration teilen',
                'with timestamp': 'Mit Zeitstempel',
                'without timestamp': 'Ohne Zeitstempel',
                'copy': 'Kopieren',
                'clear': 'Löschen',
                'edit visually': 'In YoHours-Editor öffnen',
                'yohours description': 'Ein visueller Editor für einfache Öffnungszeiten-Syntax',
                'evaluation tool': 'Auswertewerkzeug',
                'rule separator ;': 'Begrenzungszeichen für Regeln (es folgt eine normale Regel)',
                'rule separator  ||': 'Begrenzungszeichen für Regeln (es folgt eine Oder-Verknüpfte Regel die nur auf Zeiten zutrifft, die nicht bereits von vorherigen Regeln behandelt werden)',
                'rule separator ,': 'Begrenzungszeichen für Regeln (es folgt eine additive Regel deren Zeiten vorherige Regeln erweitern und nicht überschreiben wie bei normalen Regeln)',
                'JOSM remote conn error': 'Es konnte keine Verbindung zu JOSM aufgebaut werden. Bitte prüfe, ob JOSM ausgeführt wird und Fernsteuerung auf dem Standard tcp Port 8111 erlaubt ist.',
                'Affected rules': 'Betroffene Regeln',
                'Original': 'Original',
                'Comparison': 'Vergleich',
                'Rule': 'Regel',
                'Deviation at': 'Abweichung um',
                'refer to yohours': 'Dieser Wert kann auch von YoHours verarbeitet werden. YoHours ist ein benutzerfreundlicher Editor für die opening_hours Syntax. Falls keine fortschrittlichen Funktionen der Syntax benötigt werden ist YoHours eine einsteigerfreundliche Alternative. <a href="{{href}}">Ausprobieren</a>.',
                'include timestamp?': 'mit Zeitstempel?',
            },
            'words': {
                'modifier': '{{name}} Regeleigenschaft',
                'selector': '{{name}} Bereichsdefinition',
                'mode': 'Auswerte-Modus',
                'green': 'Grün',
                'yellow': 'Gelb',
                'red': 'Rot',
                'violet': 'Violett',
                'from': 'von',
                'to': 'bis',
                'and': 'und',
                'no': 'kein',
                'undefined': 'undefiniert',
                'docu': 'Dokumentation',
                'of course': 'natürlich',
                'open': 'geöffnet',
                'unknown': 'unbekannt',
                'closed': 'geschlossen',
                'comment': 'Kommentar',
                'today': 'heute',
                'tomorrow': 'morgen',
                'yesterday': 'gestern',
                'region': 'Region',
                'position': 'Position',
                'lat': 'Breitengrad',
                'lon': 'Längengrad',
                'country': 'Land',
                'state': 'Bundesstaat',
                'status': 'Status',
                'examples': 'Beispiele',
                'results': 'Ergebnisse',
                'compare': 'Vergleichen',
                'actions': 'Aktionen',
                'none': 'keine',
                'date': 'Datum',
                'time': {
                    'minute': 'Minute',
                    'minute_other': 'Minuten',
                    'minute many': 'Minuten',
                    'hour': 'Stunde',
                    'hour_other': 'Stunden',
                    'hour many': 'Stunden',
                    'day': 'Tag',
                    'day_other': 'Tage',
                    'day many': 'Tage',
                    'week': 'Woche',
                    'month': 'Monat',
                    'year': 'Jahr',
                    'year_other': 'Jahre',
                    'years many': 'Jahre',
                    'hours minutes sep': 'und ',
                    'now': 'Jetzt',
                    'time': 'Zeit',
                },
                'in duration': 'in',
            },
        },
    }, // }}}

    // Russian (ru) localization {{{
    ru: {
        translation: {
            'lang': {
                'en': 'английский',
                'fr': 'французский',
                'de': 'немецкий',
                'ru': 'русский',
                'pt': 'португальский',
                'it': 'итальянский',
                'uk': 'украинский',
                'nl': 'нидерландский',
                'hu': 'венгерский',
                'es': 'испанский',
                'vi': 'вьетнамский',
                'ja': 'Японский язык',
                'choose': 'Выберите язык',
            },

            'weekdays': {
                'day next week': 'в следующий {{day}}',
                'days next week': {
                    'su': 'в ближайшее {{day}}',
                    'mo': 'в ближайший {{day}}',
                    'tu': 'в ближайший {{day}}',
                    'we': 'в ближайшую {{day}}',
                    'th': 'в ближайший {{day}}',
                    'fr': 'в ближайшую {{day}}',
                    'sa': 'в ближайшую {{day}}',
                },
            },
            'texts': {
                'filter': {
                    'none': 'Не применять фильтры',
                    'error': 'Сообщения об ошибках и предупреждения',
                    'warnOnly': 'Только предупреждения',
                    'errorOnly': 'Только ошибки',
                    'open': 'Только заведения, которые сейчас открыты',
                    'unknown': 'Только заведения, которые могут быть открыты сейчас',
                    'closed': 'Только заведения, которые сейчас закрыты',
                    'openOrUnknown': 'Только заведения, которые открыты или могут быть открыты сейчас',
                },
                'title': 'инструмент оценки opening_hours',
                'open always': 'Заведение всегда открыто.',
                'unknown always': 'Заведение, возможно, всегда открыто.',
                'closed always': 'Заведение всегда закрыто.',
                'open now': 'Сейчас заведение открыто',
                'unknown now': 'Заведение, возможно, сейчас открыто',
                'closed now': 'Сейчас заведение закрыто',
                'will close': 'но <a href="{{href}}">закроется</a> {{timestring}}{{comment}}.',
                'will unknown': 'но <a href="{{href}}">возможно</a> откроется {{timestring}}{{comment}}.',
                'will open': 'но <a href="{{href}}">откроется</a> {{timestring}}{{comment}}.',
                'then close': 'и <a href="{{href}}">будет</a> закрыто {{timestring}}{{comment}}.',
                'then open': 'и <a href="{{href}}">будет</a> открыто {{timestring}}{{comment}}.',
                'depends on': ', но это зависит от {{comment}}.',
                'reason': 'Причина',
                'opens again': 'Откроется снова',
                'closes again': 'Закроется снова',
                'jump to time': 'Перейти к этому времени',
                'interval boundaries': 'Границы интервалов (для разработчиков)',
                'interval boundaries hint': 'Показывает все точки, где заканчивается интервал, даже без изменения состояния.',
                'upcoming changes': 'Предстоящие изменения',
                'week stable': 'Расписание верно для любой недели.',
                'not week stable': 'ВНИМАНИЕ! Расписание меняется в другие недели.',
                'value for': 'значение для',
                'value to compare': 'Сравнить с введённым значением opening_hours',
                'yohours incompatible': 'Значение несовместимо с редактором YoHours',
                'MatchingRule': 'Применённое правило',
                'prettified value': 'форматированное значение opening_hours (это значение можно безопасно использовать в OSM после <a href="{{copyFunc}}">устранения</a> всех предупреждений)',
                'prettified value for displaying': 'форматированное значение opening_hours для отображения (включая переносы строк, не используйте это значение в OSM)',
                'more information': 'Для получения дополнительной информации ознакомьтесь с <a href="{{href}}" target="_blank">вики OSM</a>.',
                'this website': 'Этот сайт и библиотека JavaScript для оценки часов работы разрабатываются на <a href="{{url}}" target="_blank">{{hoster}}</a>.',
                'if PH is between Mo and Fr': 'Только если праздничный день выпадает на будний день (Пн-Пт)',
                'check out error correction, prettify': 'проверьте исправление ошибок и функцию форматирования для значения opening_hours',
                'SH,PH or PH,SH': 'Это немного отличается от предыдущего значения. Название школьных каникул заменит названия праздников в комментарии.',
                'config POIs': 'настроить POI',
                'reload map': 'обновить карту',
                'heading map': 'Карта со слоем для тега opening_hours на основе OpenStreetMap',
                'map is showing': 'На этой карте объекты с тегом <a rel="external" href="{{wikiUrl}}">opening_hours</a> отображаются цветными кругами:',
                'error': 'Значение не удалось обработать',
                'warning': 'Если при оценке появились предупреждения, в значке статуса появится синий {{sign}}.',
                'map filter': 'Есть несколько фильтров, которые можно применить для поиска и исправления ошибок (QA) или для отображения открытых или закрытых заведений:',
                'data source': 'Данные слоя получены из {{OSMStartaTag}} и запрошены с помощью {{APIaTag}}. Карта — {{OSMaTag}}.',
                'mode 0': 'Принимаются только временные диапазоны (теги opening_hours, lit)',
                'mode 1': 'Принимаются только моменты времени',
                'mode 2': 'Принимаются временные диапазоны и моменты времени (теги service_times, collection_times)',
                'value to long for osm': 'Значение слишком длинное для OSM. База данных OpenStreetMap ограничена значениями до {{maxLength}} символов. Форматированное значение имеет длину {{pretLength}}, а введённое вами значение — {{valLength}} символов.',
                'low zoom level': 'POI начнут появляться на уровне масштабирования ${next} и выше. Текущий уровень масштабирования — ${actual}.',
                'all n entries': 'Все {{total}} записей:',
                'the first entries': 'Первые {{number}} из {{total}} записей:',
                'load all with JOSM': 'загрузить все в JOSM',
                'load osm objects': 'Загрузить все объекты OSM с этим значением opening_hours',
                'share config': 'Поделиться этой конфигурацией',
                'with timestamp': 'С временной меткой',
                'without timestamp': 'Без временной метки',
                'copy': 'Копировать',
                'clear': 'Очистить',
                'edit visually': 'Открыть в редакторе YoHours',
                'yohours description': 'Визуальный редактор для простого синтаксиса opening_hours',
                'Affected rules': 'Затронутые правила',
                'Original': 'Оригинал',
                'Comparison': 'Сравнение',
                'Rule': 'Правило',
                'Deviation at': 'Отклонение в',
                'evaluation tool': 'инструмент оценки',
                'rule separator ;': 'разделитель правил (следующее правило будет обычным правилом)',
                'rule separator  ||': 'разделитель правил (следующее правило будет резервным и применяется для времени, не охваченного предыдущими правилами)',
                'rule separator ,': 'разделитель правил (следующее правило будет дополнительным и расширяет время предыдущих правил, а не переопределяет их)',
                'JOSM remote conn error': 'Не удалось подключиться к JOSM. Убедитесь, что JOSM запущен и настроен для удалённого управления на порту TCP 8111.',
                'refer to yohours': 'Это значение также может быть обработано YoHours — простым редактором синтаксиса opening_hours. Если вам не нужны расширенные функции синтаксиса, <a href="{{href}}">попробуйте его</a>.',
                'include timestamp?': 'включить временную метку?',
            },
            'words': {
                'modifier': 'модификатор {{name}}',
                'selector': 'селектор {{name}}',
                'mode': 'режим оценки',
                'green': 'зелёный',
                'yellow': 'жёлтый',
                'red': 'красный',
                'violet': 'фиолетовый',
                'from': 'с',
                'to': 'до',
                'and': 'и',
                'no': 'Нет',
                'undefined': 'не определено',
                'docu': 'документация',
                'of course': 'конечно',
                'open': 'открыто',
                'unknown': 'неизвестно',
                'closed': 'закрыто',
                'comment': 'комментарий',
                'today': 'сегодня',
                'tomorrow': 'завтра',
                'yesterday': 'hier',
                'in duration': 'через',
                'region': 'регион',
                'position': 'положение',
                'lat': 'широта',
                'lon': 'долгота',
                'country': 'страна',
                'state': 'область',
                'status': 'статус',
                'examples': 'Примеры',
                'results': 'Результаты',
                'compare': 'Сравнить',
                'actions': 'Действия',
                'date': 'дата',
                'none': 'нет',
                'time': {
                    'minute': 'минута',
                    'minute_one': 'минуту',
                    'minute_few': 'минуты',
                    'minute_many': 'минут',
                    'minute_other': 'минут',
                    'minute many': 'минут',
                    'hour': 'час',
                    'hour_one': 'час',
                    'hour_few': 'часа',
                    'hour_many': 'часов',
                    'hour_other': 'часов',
                    'hour many': 'часов',
                    'day': 'день',
                    'day_one': 'день',
                    'day_few': 'дня',
                    'day_many': 'дней',
                    'day_other': 'дней',
                    'day many': 'дней',
                    'week': 'неделя',
                    'month': 'месяц',
                    'year': 'год',
                    'year_other': 'лет',
                    'years many': 'лет',
                    'now': 'сейчас',
                    'time': 'время',
                    'hours minutes sep': 'и ',
                },
            },
        },
    }, // }}}

    // Portuguese (pt) localization {{{
    pt: {
        translation: {
            'lang': {
                'en': 'Inglês',
                'fr': 'Francês',
                'de': 'Alemão',
                'ru': 'Russo',
                'it': 'Italiano',
                'pt': 'Português',
                'uk': 'Ucraniano',
                'nl': 'Holandês',
                'hu': 'Húngaro',
                'es': 'Espanhol',
                'vi': 'Vietnamita',
                'ja': 'Japonês',
                'choose': 'Escolha a sua linguagem',
            },

            'weekdays': {
                'day next week': 'próximo {{day}}',
                'days next week': { // palavra "próximo" antes de um dia da semana
                    'su': 'próximo {{day}}',
                    'mo': 'próxima {{day}}',
                    'tu': 'próxima {{day}}',
                    'we': 'próxima {{day}}',
                    'th': 'próxima {{day}}',
                    'fr': 'próxima {{day}}',
                    'sa': 'próximo {{day}}',
                }
            },
            'texts': {
                'filter': {
                    'none': 'Não aplicar nenhum filtro',
                    'error': 'Mensagens de erro e avisos',
                    'warnOnly': 'Somente avisos',
                    'errorOnly': 'Somente erros',
                    'open': 'Somente lugares que estão aberto agora',
                    'unknown': 'O lugar pode estar aberto agora',
                    'closed': 'Somente lugares que estão fechados agora',
                    'openOrUnknown': 'Somente lugares que estão aberto agora ou que talvez estejam',
                },
                'title': 'Ferramenta de avaliação opening_hours',
                'open always': 'O lugar está sempre aberto',
                'unknown always': 'Talvez o lugar sempre esteja aberto',
                'closed always': 'O lugar está fechado e não se sabe quando ele vai reabrir',
                'open now': 'O lugar está aberto agora',
                'unknown now': 'O lugar pode estar aberto agora',
                'closed now': 'O lugar está fechado agora',
                'will close': 'Mas <a href="{{href}}">vai</a> fechar {{timestring}}{{comment}}.',
                'will unknown': 'Mas talvez <a href="{{href}}">vá</a> abrir {{timestring}}{{comment}}.',
                'will open': 'Mas <a href="{{href}}">vai</a> abrir {{timestring}}{{comment}}.',
                'then close': 'e <a href="{{href}}">estará</a> fechado {{timestring}}{{comment}}.',
                'then open': 'e <a href="{{href}}">estará</a> aberto {{timestring}}{{comment}}.',
                'depends on': ', Mas isso depende de {{comment}}.',
                'reason': 'Motivo',
                'opens again': 'Abre novamente',
                'closes again': 'Fecha novamente',
                'jump to time': 'Ir para este horário',
                'interval boundaries': 'Limites de intervalo (para desenvolvedores)',
                'interval boundaries hint': 'Mostra todos os pontos onde um intervalo termina, mesmo sem mudança de estado.',
                'upcoming changes': 'Próximas Mudanças',
                'week stable': 'O horário é o mesmo toda semana.',
                'not week stable': 'Atenção! Este horário pode mudar em outras semanas.',
                'value for': 'Valor para a etiqueta',
                'value to compare': 'Comparar com o valor opening_hours inserido',
                'yohours incompatible': 'Valor não compatível com o editor YoHours',
                'MatchingRule': '"Substring" usada pela regra aplicada', //como traduzir?
                'prettified value': 'Valor "embelezado" da chave opening_hours (este valor pode ser utilizado no OSM sem problemas depois que todos os avisos tenham sido <a href="{{copyFunc}}">resolvidos</a>)',
                'prettified value for displaying': 'Exibição do valor "embelezado" da etiqueta "opening_hours" (inclui quebras de linha, não use isso como valor no OSM)',
                'more information': 'Para mais informações, veja a <a href="{{href}}" target="_blank">wiki do OSM</a>.',
                'this website': 'Este sítio e a biblioteca JavaScript usados para a avaliação dos horários de funcionamento são desenvolvidos no <a href="{{url}}" target="_blank">{{hoster}}</a>.',
                'if PH is between Mo and Fr': 'Somente se o feriado for um dia da semana (Mo-Fr)',
                'check out error correction, prettify': 'Verificar a correção dos erros e embelezar função para o valor da chave "opening_hours"',
                'SH,PH or PH,SH': 'Isso tem uma pequena diferença comparado com o valor anterior. Aqui os nomes dos feriados escolares vão ter prioridade sobre os nomes de outros tipos de feriados nos comentários.',
                'config POIs': 'Configurar Pontos de Interesse',
                'reload map': 'Recarregar mapa',
                'heading map': 'Mapa com uma camada para a etiqueta "opening_hours" baseado no OpenStreetMap',
                'map is showing': 'este mapa está mostrando como circulos coloridos os pontos que possuem a etiqueta <a rel="external" href="{{wikiUrl}}">opening_hours</a>:',
                'error': 'O valor não pode ser analisado',
                'warning': 'Se apareceram avisos durante a avaliação, um {{sign}} azul vai aparecer na barra de estado',
                'map filter': 'Alguns filtros podem ser aplicados para encontrar e consertar erros (CQ) ou para mostrar somente lugares abertos ou fechados:',
                'data source': 'Os dados da camada vem de {{APIaTag}}. O mapa é {{OSMaTag}}.',
                'mode 0': 'Somente intervalos de tempo são aceitos (etiquetas "opening_hours" e "lit")',
                'mode 1': 'Somente pontos específicos no tempo são aceitos',
                'mode 2': 'Intervalos de tempo e pontos específicos no tempo são aceitos (etiquetas "service_times" e "collection_times")',
                'value to long for osm': 'O valor é longo demais para o OSM. O banco de dados do OpenStreetMap atualmente é limitado a valores de até {{maxLength}} caracteres. O valor "embelezado" tem um comprimento de {{pretLength}} e o valor que você digitou tem {{valLength}} caracteres',
                'low zoom level': 'Os Pontos de Interesse vão começar a aparecer em um nível de zoom maior ou igual a ${next}. Agora você está no nível de zoom ${actual}.',
                'all n entries': 'Todos os {{total}} resultados:',
                'the first entries': 'Os primeiros {{number}} de {{total}} resultados:',
                'load all with JOSM': 'Carregar tudo no JOSM',
                'load osm objects': 'Carregar todos os objetos OSM com este valor de opening_hours',
                'share config': 'Partilhar esta configuração',
                'with timestamp': 'Com carimbo de data/hora',
                'without timestamp': 'Sem carimbo de data/hora',
                'copy': 'Copiar',
                'clear': 'Limpar',
                'edit visually': 'Abrir no editor YoHours',
                'yohours description': 'Um editor visual para sintaxe de opening_hours simples',
                'Affected rules': 'Regras afetadas',
                'Original': 'Original',
                'Comparison': 'Comparação',
                'Rule': 'Regra',
                'Deviation at': 'Desvio em',
                'evaluation tool': 'Ferramenta de avaliação',
                'rule separator ;': 'separador de regras (a próxima regra será uma regra normal)',
                'rule separator  ||': 'separador de regras (a próxima regra será uma regra de fallback que se aplica a qualquer horário não tratado pelas regras anteriores)',
                'rule separator ,': 'separador de regras (a próxima regra será uma regra adicional que estende os horários das regras anteriores sem substituí-las)',
                'JOSM remote conn error': 'Não foi possível conectar ao JOSM. Certifique-se de que o JOSM está em execução e configurado para controle remoto na porta TCP padrão 8111.',
                'refer to yohours': 'Este valor também pode ser processado pelo YoHours, um editor simples para a sintaxe opening_hours. Se você não precisa de recursos avançados da sintaxe, <a href="{{href}}">experimente</a>.',
                'include timestamp?': 'incluir carimbo de data/hora?',
            },
            'words': {
                'modifier': 'modificador {{name}}',
                'selector': 'seletor {{name}}',
                'mode': 'Modo de avaliação',
                'green': 'verde',
                'yellow': 'amarelo',
                'red': 'vermelho',
                'violet': 'violeta',
                'from': '',
                'to': 'até',
                'and': 'e',
                'no': 'sem',
                'undefined': 'indefinido',
                'docu': 'documentação',
                'of course': 'é claro',
                'open': 'Aberto de',
                'unknown': 'Desconhecido',
                'closed': 'Fechado',
                'comment': 'comentário',
                'today': 'hoje',
                'tomorrow': 'amanhã',
                'yesterday': 'ontem',
                'in duration': 'em',
                'region': 'Região',
                'position': 'Posição',
                'lat': 'Latitude',
                'lon': 'Longitude',
                'country': 'País',
                'state': 'Estado',
                'status': 'Estado',
                'examples': 'Exemplos', //revisar: maiusculo mesmo?
                'results': 'Resultados',
                'compare': 'Comparar',
                'actions': 'Ações',
                'none': 'nenhum',
                'date': 'Data',
                'time': {
                    'minute': 'minuto',
                    'minute_other': 'minutos',
                    'minute many': 'minutos',
                    'hour': 'hora',
                    'hour_other': 'horas',
                    'hour many': 'horas',
                    'day': 'dia',
                    'day_other': 'dias',
                    'day many': 'dias',
                    'week': 'semana',
                    'month': 'mês',
                    'year': 'ano',
                    'year_other': 'anos',
                    'years many': 'anos',
                    'hours minutes sep': 'e ',
                    'now': 'Agora',
                    'time': 'tempo',
                },
            },
        },
    }, // }}}

    // Italian (it) localization {{{
    it: {
        translation: {
            'lang': {
                'en': 'Inglese',
                'fr': 'Francese',
                'de': 'Tedesco',
                'ru': 'Russo',
                'it': 'Italiano',
                'pt': 'Portoghese',
                'uk': 'Ucraino',
                'nl': 'Olandese',
                'hu': 'Ungherese',
                'es': 'Spagnolo',
                'vi': 'Vietnamita',
                'ja': 'Giapponese',
                'choose': 'Scegli lingua',
            },

            'weekdays': {
                'day next week': 'il prossimo {{day}}',
                'days next week': { // The italian language use other words for next for sunday.
                    'mo': 'il prossimo {{day}}',
                    'tu': 'il prossimo {{day}}',
                    'we': 'il prossimo {{day}}',
                    'th': 'il prossimo {{day}}',
                    'fr': 'il prossimo {{day}}',
                    'sa': 'il prossimo {{day}}',
                    'su': 'la prossima {{day}}',
                }
            },
            'texts': {
                'filter': {
                    'none': 'Non applicare filtri',
                    'error': 'Messaggi di errore e di avviso',
                    'warnOnly': 'Solo avvertimenti',
                    'errorOnly': 'Solo errori',
                    'open': 'Solo strutture aperte in questo momento',
                    'unknown': 'Solo strutture che potrebbero essere aperte in questo momento',
                    'closed': 'Solo strutture che sono chiuse in questo momento',
                    'openOrUnknown': 'Solo strutture aperte o che potrebbero esserlo in questo momento',
                },
                'open always': 'La struttura è sempre aperta',
                'unknown always': 'La struttura può essere sempre aperta',
                'closed always': 'La struttura è sempre chiusa (nel breve termine)',
                'open now': 'In questo momento la struttura è aperta',
                'unknown now': 'In questo momento la struttura potrebbe essere aperta',
                'closed now': 'In questo momento la struttura è chiusa',
                'will close': 'ma <a href="{{href}}">sarà</a> chiusa {{timestring}}{{comment}}.',
                'will unknown': 'ma <a href="{{href}}">potrebbe</a> essere aperta {{timestring}}{{comment}}.',
                'will open': 'ma <a href="{{href}}">sarà</a> aperta {{timestring}}{{comment}}.',
                'then close': 'e <a href="{{href}}">sarà</a> chiusa {{timestring}}{{comment}}.',
                'then open': 'e <a href="{{href}}">sarà</a> aperta {{timestring}}{{comment}}.',
                'depends on': ', ma dipende da {{comment}}.',
                'reason': 'Motivo',
                'opens again': 'Riapre',
                'closes again': 'Richiude',
                'jump to time': 'Vai a questo orario',
                'interval boundaries': 'Limiti di intervallo (per sviluppatori)',
                'interval boundaries hint': 'Mostra tutti i punti in cui un intervallo termina, anche senza cambio di stato.',
                'upcoming changes': 'Prossimi Cambiamenti',
                'week stable': 'L\'orario è valido per ogni settimana.',
                'not week stable': 'Attenzione! Questo orario può variare nelle altre settimane.',
                'value for': 'valore per',
                'value to compare': 'Confronta con il valore opening_hours inserito',
                'yohours incompatible': 'Valore non compatibile con l\'editor YoHours',
                'MatchingRule': 'valore base utilizzato dalla regola corrente',
                'prettified value': 'valore per opening_hours ottimizzato (questo valore è valido e può essere utilizzato su OSM, dopo aver <a href="{{copyFunc}}">risolto</a> tutti gli eventuali errori)',
                'prettified value for displaying': 'valore per opening_hours ottimizzato per la visualizzazione (contiene dei caratteri di fine riga, non può essere usato come valore su OSM)',
                'more information': 'Per ulteriori informazioni è possibile consultare il <a href="{{href}}" target="_blank">wiki di OSM</a>.',
                'this website': 'Questo sito internet e la libreria Javascript utlizzata per la validazione degli orari di apertura sono sviluppati su <a href="{{url}}" target="_blank">{{hoster}}</a>.',
                'if PH is between Mo and Fr': 'Solamente se la festività pubblica è un giorno della settimana (da Lunedì a Venerdì)',
                'check out error correction, prettify': 'Controlla la correzione dell\'errore e la funzione di ottimizzazione per il valore di opening_hours',
                'SH,PH or PH,SH': 'Questo è leggermente diverso da quello precedente. Il nome delle festività scolastiche sovrascriverà quello delle festività pubbliche nei commenti.',
                'config POIs': 'configura POI',
                'reload map': 'ricarica mappa',
                'heading map': 'Mappa con livelli relativi al tag opening_hours basata su OpenStreetMap',
                'map is showing': 'questa mappa mostra con dei cerchi colorati i nodi aventi il tag <a rel="external" href="{{wikiUrl}}">opening_hours</a>:',
                'error': 'Il valore non può essere analizzato',
                'warning': 'Se sono presenti degli avvertimenti durante la validazione apparirà una {{sign}} blu nell\'icona di stato.',
                'map filter': 'Ci sono alcuni filtri che possono essere applicati per trovare e correggere gli errori o semplicemente per mostrare le strutture aperte o chiuse:',
                'data source': 'I dati provengono da {{APIaTag}}. La mappa è {{OSMaTag}}.',
                'mode 0': 'Solo intervalli di tempo (tag opening_hours, lit)',
                'mode 1': 'Solo orari precisi (es. Mo 20:00)',
                'mode 2': 'Intervalli di tempo e orari precisi (tag service_times, collection_times)',
                'value to long for osm': 'Il valore è troppo lungo per OSM. I valori del database OpenStreetMap sono attualmente limitati a {{maxLength}} caratteri. Il valore ottimizzato ha una lunghezza di {{pretLength}} e il valore inserito è lungo {{valLength}} caratteri.',
                'low zoom level': 'I POI cominceranno ad apparire ad un livello di zoom di ${next} e superiore. Il livello di zoom attuale è ${actual}.',
                'all n entries': 'Tutti i {{total}} risultati:',
                'the first entries': 'I primi {{number}} di {{total}} risultati:',
                'load all with JOSM': 'carica tutto in JOSM',
                'load osm objects': 'Carica tutti gli oggetti OSM con questo valore opening_hours',
                'share config': 'Condividi questa configurazione',
                'with timestamp': 'Con timestamp',
                'without timestamp': 'Senza timestamp',
                'Affected rules': 'Regole interessate',
                'Original': 'Originale',
                'Comparison': 'Confronto',
                'Rule': 'Regola',
                'Deviation at': 'Deviazione alle',
                'copy': 'Copia',
                'clear': 'Cancella',
                'edit visually': 'Apri nell\'editor YoHours',
                'yohours description': 'Un editor visivo per sintassi di opening_hours semplice',
                'evaluation tool': 'strumento di validazione',
                'title': 'strumento di validazione opening_hours',
                'rule separator ;': 'separatore di regole (la regola successiva sarà una regola normale)',
                'rule separator  ||': 'separatore di regole (la regola successiva sarà una regola di fallback che si applica a qualsiasi orario non gestito dalle regole precedenti)',
                'rule separator ,': 'separatore di regole (la regola successiva sarà una regola aggiuntiva che estende gli orari delle regole precedenti senza sovrascriverle)',
                'JOSM remote conn error': 'Impossibile connettersi a JOSM. Assicurarsi che JOSM sia in esecuzione e configurato per il controllo remoto sulla porta TCP predefinita 8111.',
                'refer to yohours': 'Questo valore può essere elaborato anche da YoHours, un semplice editor per la sintassi opening_hours. Se non hai bisogno delle funzionalità avanzate della sintassi, <a href="{{href}}">provalo</a>.',
                'include timestamp?': 'includere timestamp?',
            },
            'words': {
                'modifier': 'modificatore {{name}}',
                'selector': 'selettore {{name}}',
                'mode': 'modalità di validazione',
                'green': 'verde',
                'yellow': 'giallo',
                'red': 'rosso',
                'violet': 'viola',
                'from': '',
                'to': 'a',
                'and': 'e',
                'no': 'no',
                'undefined': 'non definito',
                'docu': 'documentazione',
                'of course': 'certamente',
                'open': 'aperto',
                'unknown': 'sconosciuto',
                'closed': 'chiuso',
                'comment': 'commento',
                'today': 'oggi',
                'tomorrow': 'domani',
                'yesterday': 'ieri',
                'in duration': 'tra',
                'region': 'regione',
                'position': 'posizione',
                'lat': 'latitudine',
                'lon': 'longitudine',
                'country': 'nazione',
                'state': 'stato',
                'status': 'stato',
                'examples': 'Esempi',
                'results': 'Risultati',
                'compare': 'Confronta',
                'actions': 'Azioni',
                'none': 'nessuno',
                'date': 'data',
                'time': {
                    'minute': 'minuto',
                    'minute_other': 'minuti',
                    'minute many': 'minuti',
                    'hour': 'ora',
                    'hour_other': 'ore',
                    'hour many': 'ore',
                    'day': 'giorno',
                    'day_other': 'giorni',
                    'day many': 'giorni',
                    'week': 'settimana',
                    'month': 'mese',
                    'year': 'anno',
                    'year_other': 'anni',
                    'years many': 'anni',
                    'hours minutes sep': 'e ',
                    'now': 'adesso',
                    'time': 'ora',
                },
            },
        },
    }, // }}}

    // Ukrainian (uk) localization {{{
    uk: {
        translation: {
            'lang': {
                'en': 'англійська',
                'fr': 'французька',
                'de': 'німецька',
                'ru': 'російська',
                'pt': 'португальська',
                'it': 'италійська',
                'uk': 'українська',
                'nl': 'нідерландська',
                'hu': 'угорська',
                'es': 'іспанська',
                'vi': 'в’єтнамська',
                'ja': 'Японська мова',
                'choose': 'Виберіть мову',
            },

            'weekdays': {
                'day next week': 'наступного {{day}}',
                'days next week': {
                    'su': 'у наступну {{day}}',
                    'mo': 'у наступний {{day}}',
                    'tu': 'у наступний {{day}}',
                    'we': 'у наступну {{day}}',
                    'th': 'у наступний {{day}}',
                    'fr': 'у наступну {{day}}',
                    'sa': 'у наступну {{day}}',
                },
            },
            'texts': {
                'filter': {
                    'none': 'не використовувати фільтрів',
                    'error': 'про помилки і зауваження',
                    'warnOnly': 'Тільки попередження',
                    'errorOnly': 'Тільки помилки',
                    'open': 'Тільки заходи, що зараз відчинені',
                    'unknown': 'Тільки заходи, що можуть бути зараз відчинені',
                    'closed': 'Тільки заходи, що зараз зачинені',
                    'openOrUnknown': 'Тільки заходи, що зараз відчинені або чий стан невідомий',
                },
                'open always': 'Захід завжди відчинений.',
                'closed always': 'Захід завжди зачинений.',
                'open now': 'Зараз захід відчинений',
                'closed now': 'Зараз захід зачинений',
                'will close': '<a href="{{href}}">Зачиниться</a> {{timestring}}{{comment}}.',
                'will open': '<a href="{{href}}">Відчинеться</a> {{timestring}}{{comment}}.',
                'will unknown': 'але можливо <a href="{{href}}">відчиниться</a> {{timestring}}{{comment}}.',
                'then close': 'і <a href="{{href}}">буде</a> зачинено {{timestring}}{{comment}}.',
                'then open': 'і <a href="{{href}}">буде</a> відчинено {{timestring}}{{comment}}.',
                'week stable': 'Розклад вірний для всіх тижнів.',
                'not week stable': 'УВАГА! Розклад змінюється щотижня.',
                'value for': 'значення для',
                'value to compare': 'Порівняти з введеним значенням opening_hours',
                'yohours incompatible': 'Значення несумісне з редактором YoHours',
                'MatchingRule': 'використовувати правила',

                'unknown always': 'Захід може бути завжди відчинений',
                'unknown now': 'Захід може бути зараз відкритим',
                'depends on': ', але це залежить від {{comment}}.',
                'reason': 'Причина',
                'opens again': 'Відчиниться знову',
                'closes again': 'Зачиниться знову',
                'jump to time': 'Перейти до цього часу',
                'interval boundaries': 'Межі інтервалів (для розробників)',
                'interval boundaries hint': 'Показує всі точки, де закінчується інтервал, навіть без зміни стану.',
                'upcoming changes': 'Майбутні зміни',
                'prettified value': 'припустиме значення opening_hours (цей вираз можна безпечно використовувати у OSM, після <a href="{{copyFunc}}">вирішення</a> всіх зауважень)',
                'prettified value for displaying': '\'припустиме значення opening_hours для відображення (включає кілька рядків, не використовуйте це значення для OSM)',
                'more information': 'Для додаткової інформації перейдіть на сторінку <a href="{{href}}" target="_blank">OSM wiki</a>.',
                'this website': 'Цей вебсайт та бібліотека JavaScript, що використовуються для розрахунку часів роботи, розроблена на <a href="{{url}}" target="_blank">{{hoster}}</a>.',
                'if PH is between Mo and Fr': 'Тільки якщо свято у робочий день (Пн-Пт)',
                'check out error correction, prettify': 'перевірити виправлення помилок та функцію корекції для виразу opening_hours',
                'SH,PH or PH,SH': 'Цей вираз трохи відрізняється від попереднього. Назва шкільних канікул замінить назви свят у коментарях.',
                'config POIs': 'налаштувати POI',
                'reload map': 'перезавантажити мапу',
                'heading map': 'Мапа з шаром для тегу opening_hours, на основі OpenStreetMap',
                'map is showing': 'ця мапа покакзує вузли з тегом <a rel="external" href="{{wikiUrl}}">opening_hours</a> у вигляді кольорових as colored кіл:',
                'error': 'Вираз неможливо розпізнати',
                'warning': 'Якщо Ви отримуєте попередження під час розрахунку, синій {{sign}} буде відображений на піктограмі стану.',
                'map filter': 'Є кілька фільтрів, які можна застосовувати для пошуку і виправлення помилок (QA) або просто відображення відчинених або зачинених заходів:',
                'data source': 'Накладені дані взяті з {{APIaTag}}. Мапа - {{OSMaTag}}.',
                'mode 0': 'Підтримуються тільки часові діапазони (теги opening_hours, lit)',
                'mode 1': 'Підтримуються тільки моменти часу',
                'mode 2': 'Підтримуються часові діапазони та моменти часу (теги service_times, collection_times)',
                'value to long for osm': 'Вираз занадто довгий для OSM. База даних OpenStreetMap на цей час обмежена значеннями до {{maxLength}} символів. Прийнятне значення має довжину {{pretLength}}, а Ваш вираз займає {{valLength}} символів',
                'low zoom level': 'POI почнуть з\'являтися на рівні масштабування ${next} та вище. Зараз Ви на ${actual} рівні масштабування.',
                'all n entries': 'Усі {{total}} записів:',
                'the first entries': 'Перші {{number}} з {{total}} записів:',
                'load all with JOSM': 'завантажити всі об\'єкти з цим виразом у JOSM',
                'load osm objects': 'Завантажити всі об\'єкти OSM з цим значенням opening_hours',
                'share config': 'Поділитися цією конфігурацією',
                'with timestamp': 'З часовою міткою',
                'without timestamp': 'Без часової мітки',
                'Affected rules': 'Залучені правила',
                'Original': 'Оригінал',
                'Comparison': 'Порівняння',
                'Rule': 'Правило',
                'Deviation at': 'Відхилення о',
                'copy': 'Копіювати',
                'clear': 'Очистити',
                'edit visually': 'Відкрити у редакторі YoHours',
                'yohours description': 'Візуальний редактор для простого синтаксису opening_hours',
                'evaluation tool': 'інструмент розрахунку',
                'title': 'інструмент оцінки opening_hours',
                'rule separator ;': 'роздільник правил (наступне правило буде звичайним)',
                'rule separator  ||': 'роздільник правил (наступне правило буде резервним, що застосовується до часу, не охопленого попередніми правилами)',
                'rule separator ,': 'роздільник правил (наступне правило буде додатковим і розширить час попередніх правил, не замінюючи їх)',
                'JOSM remote conn error': 'Не вдалося підключитися до JOSM. Переконайтеся, що JOSM запущено та налаштовано для віддаленого керування на TCP-порту 8111.',
                'refer to yohours': 'Це значення також може бути оброблене YoHours — простим редактором синтаксису opening_hours. Якщо вам не потрібні розширені функції синтаксису, <a href="{{href}}">спробуйте його</a>.',
                'include timestamp?': 'включити часову мітку?',
            },
            'words': {
                'modifier': 'модифікатор {{name}}',
                'selector': 'селектор {{name}}',
                'mode': 'режим розрахунку',
                'green': 'зелений',
                'yellow': 'жовтий',
                'red': 'червоний',
                'violet': 'фіолетовий',
                'from': 'з',
                'to': 'до',
                'and': 'і',
                'no': 'Нема',
                'undefined': 'не визначено',
                'docu': 'документація',
                'of course': 'звісно',
                'open': 'відкрито',
                'unknown': 'невідомо',
                'closed': 'зачинено',
                'comment': 'коментар',
                'today': 'сьогодні',
                'tomorrow': 'завтра',
                'yesterday': 'hier',
                'in duration': 'через',
                'region': 'регіон',
                'position': 'положення',
                'lat': 'широта',
                'lon': 'довгота',
                'country': 'країна',
                'state': 'область',
                'status': 'статус',
                'examples': 'Приклади',
                'results': 'Результати',
                'compare': 'Порівняти',
                'actions': 'Дії',
                'none': 'відсутні',
                'date': 'дата',
                'time': {
                    'minute': 'хвилина',
                    'minute_one': 'хвилину',
                    'minute_few': 'хвилини',
                    'minute_many': 'хвилин',
                    'minute_other': 'хвилин',
                    'minute many': 'хвилин',
                    'hour': 'година',
                    'hour_one': 'годину',
                    'hour_few': 'години',
                    'hour_many': 'годин',
                    'hour_other': 'годин',
                    'hour many': 'годин',
                    'day': 'день',
                    'day_one': 'день',
                    'day_few': 'дні',
                    'day_many': 'днів',
                    'day_other': 'днів',
                    'day many': 'днів',
                    'week': 'тиждень',
                    'month': 'місяць',
                    'year': 'рік',
                    'year_other': 'років',
                    'years many': 'років',
                    'hours minutes sep': 'та ',
                    'now': 'зараз',
                    'time': 'час',
                },
            },
        },
    }, // }}}

    // Hungarian (hu) localization {{{
    hu: {
        translation: {
            'lang': {
                'en': 'Angol',
                'fr': 'Francia',
                'de': 'Német',
                'ru': 'Orosz',
                'pt': 'Portugál',
                'it': 'Olasz',
                'uk': 'Ukrán',
                'nl': 'Holland',
                'hu': 'Magyar',
                'es': 'Spanyol',
                'vi': 'Vietnami',
                'ja': 'Japán',
                'choose': 'Nyelv kiválasztása',
            },

            'weekdays': {
                'day next week': 'következő {{day}}'
            },
            'texts': {
                'filter': {
                    'none': 'Nincs szűrő',
                    'error': 'Hibák és figyelmeztetések',
                    'warnOnly': 'Csak figyelmeztetések',
                    'errorOnly': 'Csak hibák',
                    'open': 'Csak azok a létesítmények, amelyek most nyitva vannak',
                    'unknown': 'Csak azok a létesítmények, amelyek most nyitva lehetnek',
                    'closed': 'Csak azok a létesítmények, amelyek most zárva vannak',
                    'openOrUnknown': 'Csak azok a létesítmények, amelyek most nyitva vannak vagy ismeretlen a nyitva tartásuk',
                },
                'title': 'Az [opening_hours] címke használatát értékelő eszköz',
                'open always': 'A létesítmény mindig nyitva van',
                'unknown always': 'A létesítmény mindig nyitva lehet',
                'closed always': 'A létesítmény mindig zárva van a (közeli) jövőben',
                'open now': 'A létesítmény most nyitva van',
                'unknown now': 'A létesítmény most nyitva lehet',
                'closed now': 'A létesítmény most zárva van',
                'will close': 'de <a href="{{href}}">bezár</a> ekkor: {{timestring}}{{comment}}.',
                'will unknown': 'de <a href="{{href}}">kinyithat</a> ekkor: {{timestring}}{{comment}}.',
                'will open': 'de <a href="{{href}}">kinyit</a> ekkor: {{timestring}}{{comment}}.',
                'then close': 'és <a href="{{href}}">zárva lesz</a> {{timestring}}{{comment}}.',
                'then open': 'és <a href="{{href}}">nyitva lesz</a> {{timestring}}{{comment}}.',
                'depends on': ', de ez ettől függ: {{comment}}.',
                'reason': 'Ok',
                'opens again': 'Újra nyit',
                'closes again': 'Újra zár',
                'jump to time': 'Ugrás ehhez az időponthoz',
                'interval boundaries': 'Intervallumhatárok (fejlesztőknek)',
                'interval boundaries hint': 'Minden pontot megjelenít, ahol egy intervallum véget ér, még akkor is, ha az állapot nem változik.',
                'upcoming changes': 'Közelgő Változások',
                'week stable': 'Ez az időrend minden héten érvényes.',
                'not week stable': 'Figyelem! Ez az időrend más heteken változhat.',
                'value for': 'Érték ehhez a kulcshoz:',
                'value to compare': 'Összehasonlítás a megadott opening_hours értékkel',
                'yohours incompatible': 'Az érték nem kompatibilis a YoHours szerkesztővel',
                'MatchingRule': 'Alkalmazott szabály',
                'prettified value': 'Szépen formázott [opening_hours] érték. (Biztonságosan használható az OSM-ben, miután az összes figyelmeztetés <a href="{{copyFunc}}">meg lett oldva</a>.)',
                'prettified value for displaying': 'Szépen formázott [opening_hours] érték a megjelenítéshez (sortöréseket tartalmaz, OSM értékként nem használható).',
                'more information': 'További információkért nézze meg az <a href="{{href}}" target="_blank">OSM wiki</a> szócikkeit.',
                'this website': 'Ennek a weboldalnak és a nyitva tartás kiértékeléséhez használt JavaScript programkönyvtárnak a fejlesztése a <a href="{{url}}" target="_blank">{{hoster}}</a> platformon zajlik.',
                'if PH is between Mo and Fr': 'Csak akkor, ha a munkaszüneti nap hétköznapra esik (Mo-Fr)',
                'check out error correction, prettify': 'Próbálja ki hibafelismerő eszközt és tekintse meg az opening_hours korrigált értékeit.',
                'SH,PH or PH,SH': 'Ez az érték abban különbözik az előzőtől, hogy a megjegyzésben a tanítási szünetek neve felülírja a munkaszüneti napok nevét, ha átfedik egymást.',
                'config POIs': 'POI-k beállítása',
                'reload map': 'Térkép újratöltése',
                'heading map': 'OpenStreetMap-alapú térkép az [opening_hours] címke használatának kiértékeléséhez',
                'map is showing': 'Ez a térkép az <a rel="external" href="{{wikiUrl}}">[opening_hours]</a> címkével rendelkező objektumokat jeleníti meg színes körökkel:',
                'error': 'A nyitva tartás kiértékelhetetlen',
                'warning': 'Ha figyelmeztetés jelenik meg a kiértékelés közben, akkor az állapotikonban egy kék {{sign}} látható.',
                'map filter': 'Az érdeklődésre számot tartó nyitva tartási időket szűrőkkel lehet megjeleníteni. A szűrőket alkalmazhatjuk minőségbiztosítási (hibakeresési) céllal vagy egyszerűen azért, hogy nyitva tartó üzleteket találjunk:',
                'data source': 'A kiértékelendő adatok az {{OSMStartaTag}} adatbázisából származnak, és az {{APIaTag}} segítségével vannak lekérdezve. A térkép szerzői jogainak birtokosai: {{OSMaTag}}.',
                'mode 0': 'Csak időtartományok fogadhatók el ([opening_hours] és [lit] címkék)',
                'mode 1': 'Csak időpontok fogadhatók el',
                'mode 2': 'Időtartományok és időpontok egyaránt elfogadhatók ([service_times] és [collection_times] címkék)',
                'value to long for osm': 'Az érték túl hosszú az OSM számára. Az OpenStreetMap adatbázisa jelenleg legfeljebb {{maxLength}} karakter hosszúságú értékeket támogat. A szépen formázott érték {{pretLength}} hosszúságú, a megadott érték viszont {{valLength}} karakteres.',
                'low zoom level': 'A jelölők csak ${next} nagyítási szinttől kezdenek megjelenni. A  jelenlegi nagyítási szint: ${actual}.',
                'all n entries': 'Az összes ({{total}}) bejegyzés:',
                'the first entries': 'Az első {{number}} bejegyzés (összes: {{total}}):',
                'load all with JOSM': 'Az összes betöltése a JOSM-ban',
                'load osm objects': 'Az összes OSM objektum betöltése ezzel az opening_hours értékkel',
                'share config': 'Beállítás megosztása',
                'with timestamp': 'Időbélyeggel',
                'Affected rules': 'Érintett szabályok',
                'Original': 'Eredeti',
                'Comparison': 'Összehasonlítás',
                'Rule': 'Szabály',
                'Deviation at': 'Eltérés',
                'without timestamp': 'Időbélyeg nélkül',
                'copy': 'Másolás',
                'clear': 'Törlés',
                'edit visually': 'Megnyitás YoHours-szerkesztőben',
                'yohours description': 'Vizuális szerkesztő egyszerű opening_hours szintaxishoz',
                'evaluation tool': 'Kiértékelő eszköz',
                'rule separator ;': 'Szabályelválasztó (utána egy rendes szabály következik)',
                'rule separator  ||': 'Szabályelválasztó (utána egy tartalékszabály [vagylagos szabály] következik, amely csak arra az időre vonatkozik, amelyiket az előző szabály nem érinti)',
                'rule separator ,': 'Szabályelválasztó (utána egy kiegészítő szabály kövektezik, amely – a rendes szabállyal ellentétben – nem írja felül az előző szabályok idejeit, csupák kiterjeszti azokat)',
                'JOSM remote conn error': 'Nem sikerült kapcsolódni a JOSM-hoz. Győződjön meg arról, hogy a JOSM fut-e, és be van-e állítva az alapértelmezett 8111-es TCP porton keresztüli távoli vezérlés (távirányító).',
                'refer to yohours': 'Ezt az értéket fel tudja dolgozni a YoHours is, egy felhasználóbarát címkeszerkesztő az opening_hours szintaxisához. Ha nincs szüksége a szintaxis speciális funkcióira, akkor YoHours remek eszköz. <a href="{{href}}">Próbálja ki!</a>',
                'include timestamp?': 'tartalmazzon-e időbélyeget?',
            },
            'words': {
                'modifier': '{{name}} módosító',
                'selector': '{{name}} szelektor',
                'mode': 'Kiértékelési mód',
                'green': 'zöld',
                'yellow': 'sárga',
                'red': 'piros',
                'violet': 'lila',
                'from': '',
                'to': 'eddig',
                'and': 'és',
                'no': 'nincs',
                'undefined': 'meghatározatlan',
                'docu': 'dokumentáció',
                'of course': 'természetesen',
                'open': 'nyitva',
                'unknown': 'ismeretlen',
                'closed': 'zárva',
                'comment': 'megjegyzés',
                'today': 'ma',
                'tomorrow': 'holnap',
                'yesterday': 'tegnap',
                'in duration': 'ennyi idő múlva:',
                'region': 'régió',
                'position': 'Pozíció',
                'lat': 'szélesség',
                'lon': 'hosszúság',
                'country': 'ország',
                'state': 'tartomány',
                'status': 'Állapot',
                'examples': 'Példák',
                'results': 'Eredmények',
                'compare': 'Összehasonlítás',
                'actions': 'Műveletek',
                'none': 'nincs',
                'date': 'Dátum',
                'time': {
                    'minute': 'perc',
                    'minute_other': 'perc',
                    'minute many': 'perc',
                    'hour': 'óra',
                    'hour_other': 'óra',
                    'hour many': 'óra',
                    'day': 'nap',
                    'day_other': 'nap',
                    'day many': 'nap',
                    'week': 'hét',
                    'month': 'hónap',
                    'year': 'év',
                    'year_other': 'év',
                    'years many': 'év',
                    'hours minutes sep': 'és ',
                    'now': 'most',
                    'time': 'idő',
                },
            },
        },
    }, // }}}

    // Vietnamese (vi) localization {{{
    vi: {
        translation: {
            'lang': {
                'en': 'Anh',
                'fr': 'Pháp',
                'de': 'Đức',
                'ru': 'Nga',
                'pt': 'Bồ Đào Nha',
                'it': 'Ý',
                'uk': 'Ukraina',
                'nl': 'Hà Lan',
                'hu': 'Hungary',
                'es': 'Tây Ban Nha',
                'vi': 'Việt',
                'ja': 'Nhật Bản ngữ',
                'choose': 'Chọn Ngôn ngữ',
            },

            'weekdays': {
                'day next week': 'vào {{day}} tuần sau'
            },
            'texts': {
                'filter': {
                    'none': 'Không áp dụng bộ lọc nào',
                    'error': 'Thông báo lỗi và cảnh báo',
                    'warnOnly': 'Chỉ cảnh báo',
                    'errorOnly': 'Chỉ lỗi',
                    'open': 'Chỉ các nơi đang mở cửa',
                    'unknown': 'Chỉ các nơi có thể đang mở cửa',
                    'closed': 'Chỉ các nơi đang dóng cửa',
                    'openOrUnknown': 'Chỉ các nơi đang mở cửa hoặc không rõ',
                },
                'title': 'Công cụ đánh giá opening_hours',
                'open always': 'Luôn mở cửa',
                'unknown always': 'Luôn có thể mở cửa',
                'closed always': 'Luôn đóng cửa trong tương lai (gần đây)',
                'open now': 'Đang mở cửa',
                'unknown now': 'Có thể đang mở cửa',
                'closed now': 'Đang đóng cửa',
                'will close': 'nhưng <a href="{{href}}">sẽ</a> đóng cửa {{timestring}}{{comment}}.',
                'will unknown': 'nhưng có thể <a href="{{href}}">sẽ</a> mở cửa {{timestring}}{{comment}}.',
                'will open': 'nhưng <a href="{{href}}">sẽ</a> mở cửa {{timestring}}{{comment}}.',
                'then close': 'và <a href="{{href}}">sẽ</a> đóng cửa {{timestring}}{{comment}}.',
                'then open': 'và <a href="{{href}}">sẽ</a> mở cửa {{timestring}}{{comment}}.',
                'depends on': ', tùy theo {{comment}}.',
                'reason': 'Lý doMở cửa lại',
                'closes again': 'Đóng cửa lại',
                'jump to time': 'Đi đến thời gian này',
                'interval boundaries': 'Ranh giới khoảng thời gian (cho nhà phát triển)',
                'interval boundaries hint': 'Hiển thị tất cả các điểm mà một khoảng thời gian kết thúc, ngay cả khi không có thay đổi trạng thái.',
                'upcoming changes': 'Thay đổi sắp tới',
                'week stable': 'Thời biểu nhất định theo tuần lễ.',
                'not week stable': 'Chú ý! Thời biểu này có thể thay đổi tùy theo tuần lễ.',
                'value for': 'giá trị',
                'value to compare': 'So sánh với giá trị opening_hours đã nhập',
                'yohours incompatible': 'Giá trị không tương thích với trình chỉnh sửa YoHours',
                'MatchingRule': 'Quy tắc được áp dụng',
                'prettified value': 'Giá trị opening_hours làm đẹp (có thể an toàn sử dụng giá trị này trong OSM sau khi <a href="{{copyFunc}}">khắc phục</a> các cảnh báo)',
                'prettified value for displaying': 'Giá trị opening_hours làm đẹp để hiển thị (có nhiều dòng, đừng sử dụng giá trị này trong OSM)',
                'more information': 'Để biết thêm thông tin, xem <a href="{{href}}" target="_blank">wiki OSM</a>.',
                'this website': 'Trang Web này và thư viện JavaScript đánh giá giờ mở cửa do <a href="{{url}}" target="_blank">{{hoster}}</a> phục vụ.',
                'if PH is between Mo and Fr': 'Chỉ khi ngày nghỉ lễ công cộng là ngày thường (thứ 2–thứ 5)',
                'check out error correction, prettify': 'kiểm tra gợi ý sửa lỗi và hàm làm đẹp cho giá trị opening_hours',
                'SH,PH or PH,SH': 'Thứ tự này hơi khác với giá trị bên trên. Các tên ngày nghỉ lễ dành cho học sinh và sinh viên sẽ ghi đè các tên ngày nghỉ lễ công cộng trong lời ghi chú.',
                'config POIs': 'thiết lập các địa điểm quan tâm',
                'reload map': 'tải lại bản đồ',
                'heading map': 'Bản đồ có lớp ghim ứng với các thẻ opening_hours trong OpenStreetMap',
                'map is showing': 'Bản đồ hiển thị các đối tượng có thẻ <a rel="external" href="{{wikiUrl}}">opening_hours</a> như điểm tròn tô màu:',
                'error': 'Không thể phân tích giá trị',
                'warning': 'Các cảnh báo khi phân tích sẽ xuất hiện như {{sign}} màu xanh trong hình tượng trạng thái.',
                'map filter': 'Có thể áp dụng một số bộ lọc để phát hiện và khắc phục các lỗi (BĐCL) hoặc chỉ hiển thị các nơi đang mở cửa hoặc đóng cửa:',
                'data source': 'Dữ liệu lớp phủ bắt nguồn từ {{OSMStartaTag}} và được truy vấn dùng {{APIaTag}}. Bản đồ là {{OSMaTag}}.',
                'mode 0': 'Chỉ cho phép thời khoảng (các thẻ opening_hours, lit)',
                'mode 1': 'Chỉ cho phép thời điểm',
                'mode 2': 'Cho phép cả thời khoảng lẫn thời điểm (các thẻ service_times, collection_times)',
                'value to long for osm': 'Giá trị quá dài đối với OSM. Cơ sở dữ liệu OpenStreetMap chỉ hỗ trợ các giá trị dài cho tới {{maxLength}} ký tự. Giá trị làm đẹp có {{pretLength}} ký tự, trong khi giá trị được nhập có {{valLength}} ký tự',
                'low zoom level': 'Các địa điểm quan tâm sẽ bắt đầu xuất hiện ở mức thu phóng ${next} trở lên. Bạn đang xem mức thu phóng ${actual}.',
                'all n entries': 'Tổng số {{total}} mục:',
                'the first entries': '{{number}} mục đầu tiên trên tổng số {{total}} mục:',
                'load all with JOSM': 'tải toàn bộ trong JOSM',
                'load osm objects': 'Tải tất cả các đối tượng OSM với giá trị opening_hours này',
                'share config': 'Chia sẻ cấu hình này',
                'with timestamp': 'Với dấu thời gian',
                'Affected rules': 'Quy tắc bị ảnh hưởng',
                'Original': 'Gốc',
                'Comparison': 'So sánh',
                'Rule': 'Quy tắc',
                'Deviation at': 'Khác biệt tại',
                'without timestamp': 'Không có dấu thời gian',
                'copy': 'Sao chép',
                'clear': 'Xóa',
                'edit visually': 'Mở trong trình chỉnh sửa YoHours',
                'yohours description': 'Trình chỉnh sửa trực quan cho cú pháp opening_hours đơn giản',
                'evaluation tool': 'công cụ đánh giá',
                'rule separator ;': 'ký tự phân tách quy tắc (quy tắc sau sẽ là quy tắc bình thường)',
                'rule separator  ||': 'ký tự phân tách quy tắc (quy tắc sau sẽ là quy tắc dự trữ ứng với các thời gian bên ngoài các quy tắc trước)',
                'rule separator ,': 'ký tự phân tách quy tắc (quy tắc sau sẽ là quy tắc bổ sung để kéo dài các quy tắc trước mà không ghi đè các quy tắc trước, khác với quy tắc bình thường)',
                'JOSM remote conn error': 'Không thể kết nối với JOSM. Xin vui lòng kiểm tra rằng JOSM đang chạy và được thiết lập để điều khiển từ xa qua cổng mặc định 8111 của TCP.',
                'refer to yohours': 'YoHours cũng có thể phân tích giá trị này, đây là một trình soạn thảo cú pháp opening_hours đơn giản. Nếu bạn không cần các cú pháp nâng cao, <a href="{{href}}">hãy thử YoHours</a>.',
                'include timestamp?': 'bao gồm dấu thời gian?',
            },
            'words': {
                'modifier': 'bộ thay đổi {{name}}',
                'selector': 'bộ chọn {{name}}',
                'mode': 'chế độ đánh giá',
                'green': 'xanh',
                'yellow': 'vàng',
                'red': 'đỏ',
                'violet': 'tím',
                'from': 'từ',
                'to': 'đến',
                'and': 'và',
                'no': 'không',
                'undefined': 'không định rõ',
                'docu': 'tài liệu',
                'of course': 'dĩ nhiên',
                'open': 'mở cửa',
                'unknown': 'không rõ',
                'closed': 'đóng cửa',
                'comment': 'ghi chú',
                'today': 'hôm nay',
                'tomorrow': 'ngày mai',
                'yesterday': 'hôm qua',
                'in duration': 'còn',
                'region': 'khu vực',
                'position': 'địa điểm',
                'lat': 'vĩ độ',
                'lon': 'kinh độ',
                'country': 'quốc gia',
                'state': 'tỉnh/tiểu bang',
                'status': 'trạng thái',
                'examples': 'Ví dụ',
                'results': 'Kết quả',
                'compare': 'So sánh',
                'actions': 'Hành động',
                'none': 'không có',
                'date': 'ngày',
                'time': {
                    'minute': 'phút',
                    'minute_other': 'phút',
                    'minute many': 'phút',
                    'hour': 'giờ',
                    'hour_other': 'giờ',
                    'hour many': 'giờ',
                    'day': 'ngày',
                    'day_other': 'ngày',
                    'day many': 'ngày',
                    'week': 'tuần',
                    'month': 'tháng',
                    'year': 'năm',
                    'year_other': 'năm',
                    'years many': 'năm',
                    'hours minutes sep': ' ',
                    'now': 'bây giờ',
                    'time': 'giờ',
                },
            },
        },
    }, // }}}

        // Japanese (ja) localization {{{
        ja: {
            translation: {
                'lang': {
                    'en': '英語',
                    'fr': 'フランス語',
                    'de': 'ドイツ語',
                    'ru': 'ロシア語',
                    'pt': 'ポルトガル語',
                    'it': 'イタリア語',
                    'uk': 'ウクライナ語',
                    'nl': 'オランダ語',
                    'hu': 'ハンガリー語',
                    'es': 'スペイン語',
                    'vi': 'ベトナム語',
                    'ja': '日本語',
                    'choose': '言語の選択',
                },

                'weekdays': {
                    'day next week': 'next {{day}}'
                },
                'texts': {
                    'filter': {
                        'none': 'フィルターを適用しない',
                        'error': 'エラーと警告',
                        'warnOnly': '警告のみ',
                        'errorOnly': 'エラーのみ',
                        'open': '現在営業している施設のみ',
                        'unknown': '現在営業している可能性のある施設のみ',
                        'closed': '現在休止中の施設のみ',
                        'openOrUnknown': '現在オープンしているまたは不明な施設のみ',
                    },
                    'title': 'opening_hours（営業時間） 評価ツール',
                    'open always': 'この施設は年中無休で営業しています',
                    'unknown always': 'この施設はおそらく年中無休で営業しています',
                    'closed always': 'この施設は常時休止中です',
                    'open now': '現在の時刻ではこの施設は営業中です',
                    'unknown now': '現在の時刻ではこの施設は営業中です',
                    'closed now': '現在の時刻ではこの施設は休止中です',
                    'will close': 'ただし <a href="{{href}}">{{timestring}}</a> に休止します{{comment}}',
                    'will unknown': 'ただしおそらく <a href="{{href}}">{{timestring}}</a> に営業を開始します{{comment}}',
                    'will open': 'ただし <a href="{{href}}">{{timestring}}</a> に営業を開始します{{comment}}',
                    'depends on': '。 ただし {{comment}} によります。',
                    'week stable': 'このスケジュールはどの週でも有効です。',
                    'not week stable': '※注意 このスケジュールは他の週の場合変更される可能性があります。',
                    'value for': '',
                    'value to compare': '入力されたopening_hours値と比較',
                    'yohours incompatible': '値はYoHoursエディタと互換性がありません',
                    'MatchingRule': '適用されているルール',
                    'prettified value': '整形されたopening_hoursの値 (この値は、すべての警告が<a href="{{copyFunc}}">修正</a>された後、OSM で問題なく使用できます)',
                    'prettified value for displaying': '表示用に整形されたopening_hoursの値(改行を含みます、OSMでは使用しないでください)',
                    'more information': '詳しい情報は<a href="{{href}}" target="_blank">OSM wiki</a>で確認できます。',
                    'this website': '本サイトおよびopening_hoursの評価に使用されているJavaScriptライブラリは<a href="{{url}}" target="_blank">{{hoster}}</a>で開発されています。',
                    'if PH is between Mo and Fr': '祝日が平日の場合のみ(Mo-Fr)',
                    'check out error correction, prettify': 'エラー修正機能とopening_hoursの値の整形機能の確認',
                    'SH,PH or PH,SH': 'これは上の値とわずかな違いがあります。SHはPHより優先されます。',
                    'config POIs': '表示するPOIの設定',
                    'reload map': 'マップを再読み込み',
                    'heading map': 'OpenStreetMap opening_hoursタグの分布・使用状況マップ',
                    'map is showing': 'この地図は、<a rel="external" href="{{wikiUrl}}">opening_hours</a>というタグが付いたオブジェクトを色付きの円で表示しています。',
                    'error': '値を解読できませんでした',
                    'warning': '評価中に警告が表示された場合、ステータスアイコンに青い{{sign}}が表示されます。',
                    'map filter': 'また、いくつかのフィルターを適用することで、間違いを見つけて修正したり（QA）、営業している施設や閉じている施設を表示したりすることができます。',
                    'data source': 'オーバーレイデータは{{OSMStartaTag}}から取得し、{{APIaTag}}を使用して検索しています。地図は{{OSMaTag}}です。',
                    'mode 0': '範囲指定された時間のみ（opening_hours, litタグ）',
                    'mode 1': '時点のみ',
                    'mode 2': '範囲指定された時間と時点（service_times, collection_times タグ）。',
                    'value to long for osm': 'この値はOSMで使用するには長すぎます。現状、OpenStreetMapデータベースの値は、{{maxLength}}文字までに制限されています。整形された値の長さは{{pretLength}}で、入力された値の長さは{{valLength}}文字です。',
                    'low zoom level': 'POIは、ズームレベル${next}以上で表示されるようになります。現在のズームレベルは${actual}です。',
                    'all n entries': '全{{total}}つのエントリー:',
                    'Affected rules': '影響を受けるルール',
                    'Original': '元',
                    'Comparison': '比較',
                    'Rule': 'ルール',
                    'Deviation at': '差異',
                    'the first entries': '最初の{{total}}中{{number}}のエントリー:',
                    'load all with JOSM': 'JOSMで読み込む',
                    'load osm objects': 'このopening_hours値を持つすべてのOSMオブジェクトを読み込む',
                    'share config': 'この設定を共有',
                    'with timestamp': 'タイムスタンプあり',
                    'without timestamp': 'タイムスタンプなし',
                    'copy': 'コピー',
                    'clear': 'クリア',
                    'edit visually': 'YoHoursエディターで開く',
                    'yohours description': 'シンプルなopening_hours構文用のビジュアルエディタ',
                    'evaluation tool': '評価ツール',
                    'rule separator ;': 'ルールセパレータ（次のルールは通常のルールになります）',
                    'rule separator  ||': 'ルールセパレータ (次のルールは、前のルールで処理されなかった時間に適用されるフォールバックルールになります)',
                    'rule separator ,': 'ルールセパレータ (次のルールは、前のルールの時間を延長する追加ルールで、通常のルールのように上書きされることはありません)',
                    'JOSM remote conn error': 'JOSMに接続できませんでした。JOSMが起動していること、デフォルトのtcpポート8111でリモートコントロールができるように設定されていることを確認してください。',
                    'refer to yohours': 'この値は、opening_hours構文用のシンプルなエディタであるYoHoursでも解析できます。もし構文の高度な機能を必要としないのであれば<a href="{{href}}">試してみてください。</a>',
                    'include timestamp?': 'タイムスタンプを含めるか',
                },
                'words': {
                    'modifier': '{{name}} 修飾子',
                    'selector': '{{name}} セレクター',
                    'mode': '評価モード',
                    'green': '緑',
                    'yellow': '黄色',
                    'red': '赤',
                    'violet': '紫',
                    'from': '',
                    'to': '～',
                    'and': '･',
                    'no': 'なし',
                    'undefined': '未定義',
                    'docu': '文書',
                    'of course': 'もちろん',
                    'open': '営業中',
                    'unknown': '不明',
                    'closed': '休止中',
                    'comment': '(コメント)',
                    'today': '今日',
                    'tomorrow': '明日',
                'yesterday': 'hier',
                    'in duration': 'あと',
                    'region': '地域',
                    'position': '位置',
                    'lat': '緯度',
                    'lon': '経度',
                    'country': '国',
                    'state': '州',
                    'status': 'ステータス',
                    'examples': '例',
                    'results': '結果',
                    'compare': '比較',
                    'actions': 'アクション',
                    'none': 'なし',
                    'date': '日付',
                    'time': {
                        'minute': '分後',
                        'minute_other': '分後',
                        'minute many': '分後',
                        'hour': '時間',
                        'hour_other': '時間',
                        'hour many': '時間',
                        'day': '日',
                        'day_other': '日',
                        'day many': '日',
                        'week': '週間',
                        'month': '月',
                        'year': '年',
                        'year_other': '年',
                        'years many': '年',
                        'hours minutes sep': '',
                        'now': '現在の時刻',
                        'time': '時間',
                    },
                },
            },
        }, // }}}
};

// Functions which generate localized HTML sections {{{

export function changeLanguage(lang) {
    localStorage.setItem('i18nextLng', lang);
    i18next.changeLanguage(lang, () => {
        window.location.reload();
    });
}

export function getUserSelectTranslateHTMLCode() {
    let res = '<label for="language-select" class="hd">';
    res += i18next.t('lang.choose')
            + (i18next.language !== 'en' ? ' ('+ i18next.t('lang.choose', { lng: 'en' }) +')' : '' )
            + ':</label> ';
    res += '<select id="language-select">';
    for (const lang in resources) {
        if (Object.prototype.hasOwnProperty.call(resources, lang)) {
          const selected = i18next.language === lang ? ' selected' : '';
          res += '<option value="' + lang + '"' + selected + '>'
                + (i18next.language === 'en'
                    ? i18next.t('lang.' + lang)
                    : i18next.t('lang.' + lang, { lng: 'en' }) + ' (' + i18next.t('lang.' + lang) + ')'
                  )
                + '</option>';
        }
    }
    res += '</select>';
    return res;
}
// }}}

// Detect language: localStorage > browser language > fallback
export function detectLanguage() {
    const stored = localStorage.getItem('i18nextLng');
    if (stored && resources[stored]) return stored;

    for (const lang of navigator.languages) {
        const shortLang = lang.split('-')[0];
        if (resources[shortLang]) return shortLang;
    }
    return 'en';
}
// }}}
// }}}
